diff -ru dpkg-1.8.3.1/archtable dpkg-1.8.3.1-patched/archtable
--- dpkg-1.8.3.1/archtable	Fri Jan 12 14:55:49 2001
+++ dpkg-1.8.3.1-patched/archtable	Fri Mar 30 18:38:53 2001
@@ -43,4 +43,6 @@
 s390-linux-gnu		s390		s390
 s390-ibm-linux-gnu	s390		s390
 s390-unknown-linux-gnu	s390		s390
+powerpc-darwin		darwin-powerpc	darwin-powerpc
+i386-darwin		darwin-i386	darwin-i386
 
diff -ru dpkg-1.8.3.1/lib/Makefile.in dpkg-1.8.3.1-patched/lib/Makefile.in
--- dpkg-1.8.3.1/lib/Makefile.in	Sun May 21 19:22:34 2000
+++ dpkg-1.8.3.1-patched/lib/Makefile.in	Fri Mar 30 18:38:53 2001
@@ -16,9 +16,10 @@
 
 .PHONY: all
 all:: libdpkg.a
-	
+
 libdpkg.a: $(OBJECTS)
 	$(AR) r $@ $^
+	ranlib $@
 
 .PHONY: install
 install:: all
diff -ru dpkg-1.8.3.1/main/configure.c dpkg-1.8.3.1-patched/main/configure.c
--- dpkg-1.8.3.1/main/configure.c	Thu Aug 24 22:34:56 2000
+++ dpkg-1.8.3.1-patched/main/configure.c	Fri Mar 30 18:38:53 2001
@@ -340,8 +340,10 @@
           " I do not mess up your careful work.\n"),
                     cdr.buf, cdr2.buf);
 
+#ifndef __APPLE__
             s= getenv(NOJOBCTRLSTOPENV);
             if (s && *s) {
+#endif
               fputs(_("Type `exit' when you're done.\n"),stderr);
               if (!(c1= m_fork())) {
                 s= getenv(SHELLENV);
@@ -351,11 +353,13 @@
               }
               while ((r= waitpid(c1,&status,0)) == -1 && errno == EINTR);
               if (r != c1) { onerr_abort++; ohshite(_("wait for shell failed")); }
+#ifndef __APPLE__
             } else {
               fputs(_("Don't forget to foreground (`fg') this "
                     "process when you're done !\n"),stderr);
               kill(-getpgid(0),SIGTSTP);
             }
+#endif
           }
 
         } while (!strchr("yino",cc));
diff -ru dpkg-1.8.3.1/main/help.c dpkg-1.8.3.1-patched/main/help.c
--- dpkg-1.8.3.1/main/help.c	Fri Jan 12 17:02:12 2001
+++ dpkg-1.8.3.1-patched/main/help.c	Fri Mar 30 18:40:26 2001
@@ -69,7 +69,7 @@
 
 void checkpath(void) {
 /* Verify that some programs can be found in the PATH. */
-  static const char *const checklist[]= { "ldconfig", 
+  static const char *const checklist[]= {
 #ifdef USE_START_STOP_DAEMON
     "start-stop-daemon",
 #endif    
diff -ru dpkg-1.8.3.1/optlib/Makefile.in dpkg-1.8.3.1-patched/optlib/Makefile.in
--- dpkg-1.8.3.1/optlib/Makefile.in	Wed Dec 15 16:29:16 1999
+++ dpkg-1.8.3.1-patched/optlib/Makefile.in	Fri Mar 30 18:38:53 2001
@@ -11,9 +11,10 @@
 
 .PHONY: all
 all:: libopt.a
-	
+
 libopt.a: $(OBJECTS)
 	$(AR) r $@ $^
+	ranlib $@
 
 .PHONY: install
 install:: all
diff -ru dpkg-1.8.3.1/scripts/dpkg-architecture.pl dpkg-1.8.3.1-patched/scripts/dpkg-architecture.pl
--- dpkg-1.8.3.1/scripts/dpkg-architecture.pl	Fri Jan 12 14:56:09 2001
+++ dpkg-1.8.3.1-patched/scripts/dpkg-architecture.pl	Fri Mar 30 18:38:53 2001
@@ -61,7 +61,9 @@
 	    'hppa',      'hppa-linux',
 	    'hurd-i386', 'i386-gnu',
 	    's390',	 's390-linux',
-	    'freebsd-i386', 'i386-freebsd');
+	    'freebsd-i386', 'i386-freebsd',
+	    'darwin-powerpc', 'powerpc-darwin',
+	    'darwin-i386', 'i386-darwin');
 
 sub usageversion {
     print STDERR
