diff -ru gdk-pixbuf-0.8.0/configure.in gdk-pixbuf-0.8.0-patched/configure.in
--- gdk-pixbuf-0.8.0/configure.in	Mon May 22 18:29:21 2000
+++ gdk-pixbuf-0.8.0-patched/configure.in	Sun Jan 14 17:48:17 2001
@@ -137,7 +137,7 @@
           AC_CHECK_HEADER(tiffio.h,
             TIFF='tiff'; LIBTIFF='-ltiff34 -ljpeg -lz',
             AC_MSG_WARN(*** TIFF loader will not be built (TIFF header files not found) ***)),
-        AC_MSG_WARN(*** TIFF plug-in will not be built (TIFF library not found) ***), -ljpeg -lz -lm), -ljpeg -lz -lm), -lm)
+        AC_MSG_WARN(*** TIFF plug-in will not be built (TIFF library not found) ***), -ljpeg -lz), -ljpeg -lz), )
   fi
 
 dnl Test for libjpeg
@@ -173,7 +173,7 @@
       AC_CHECK_HEADER(png.h,
         png_ok=yes,
         png_ok=no),
-      AC_MSG_WARN(*** PNG loader will not be built (PNG library not found) ***), -lz -lm)
+      AC_MSG_WARN(*** PNG loader will not be built (PNG library not found) ***), -lz)
     if test "$png_ok" = yes; then
       AC_MSG_CHECKING([for png_structp in png.h])
       AC_TRY_COMPILE([#include <png.h>],
