diff -ru gettext-0.10.35/intl/Makefile.in gettext-0.10.35-patched/intl/Makefile.in
--- gettext-0.10.35/intl/Makefile.in	Mon Apr 27 23:53:18 1998
+++ gettext-0.10.35-patched/intl/Makefile.in	Wed Jan 31 21:50:38 2001
@@ -41,7 +41,7 @@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS = @MKINSTALLDIRS@
 
-l = @l@
+l = l
 
 AR = ar
 CC = @CC@
@@ -94,6 +94,7 @@
 libintl.la: $(OBJECTS)
 	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ $(OBJECTS) \
 		   -version-info 1:0 -rpath $(libdir)
+	ln -s .libs/libintl.a libintl.a
 
 ../po/cat-id-tbl.$lo: ../po/cat-id-tbl.c $(top_srcdir)/po/$(PACKAGE).pot
 	cd ../po && $(MAKE) cat-id-tbl.$lo
@@ -119,7 +120,11 @@
 	    $(top_srcdir)/mkinstalldirs $(libdir) $(includedir); \
 	  fi; \
 	  $(INSTALL_DATA) intlh.inst $(includedir)/libintl.h; \
-	  $(INSTALL_DATA) libintl.a $(libdir)/libintl.a; \
+	  if test -r libintl.la; then \
+	    $(LIBTOOL) --mode=install $(INSTALL_DATA) libintl.la $(libdir)/libintl.la; \
+	  else \
+	    $(INSTALL_DATA) libintl.a $(libdir)/libintl.a; \
+	  fi; \
 	else \
 	  : ; \
 	fi
