--- gnome-core-1.2.4/applets/desk-guide/gwmthumbnail.c	Wed Nov 15 18:57:58 2000
+++ gnome-core-1.2.4-patched/applets/desk-guide/gwmthumbnail.c	Mon Jan 15 20:34:48 2001
@@ -29,23 +29,29 @@
 
 
 /* --- Gdk & GdkPixbuf compat prototypes --- */
+static
 GdkImage*       gdk_image_get                    (GdkWindow *window,
 						  gint       x,
 						  gint       y,
 						  gint       width,
 						  gint       height);
+static
 GdkColormap*	gdk_window_get_colormap		 (GdkWindow *window);
+static
 GdkVisual*	gdk_window_get_visual		 (GdkWindow *window);
+static
 GdkImage*       gdk_image_new_shared_with_pixmap (GdkWindow  *window,
 						  gint        width,
 						  gint        height,
 						  GdkPixmap **pixmap_p);
+static
 GdkPixbuf*      gdk_pixbuf_from_image            (GdkImage    *image,
 						  guint        x,
 						  guint        y,
 						  guint        width,
 						  guint        height,
 						  GdkColormap *cmap);
+static
 void            gdk_pixbuf_copy_from_image       (GdkPixbuf   *pixbuf,
 						  guint        dest_x,
 						  guint        dest_y,
@@ -537,6 +543,7 @@
 #include <sys/shm.h>
 #endif	/* USE_SHM */
 
+static
 GdkImage*
 gdk_image_new_shared_with_pixmap (GdkWindow  *window,
 				  gint        width,
@@ -616,6 +623,7 @@
 
 /* fixed gdk_image_get(), so we don't crash if we get an X error
  */
+static
 GdkImage*
 gdk_image_get (GdkWindow *window,
 	       gint       x,
@@ -676,6 +684,7 @@
 /* instead of bailing on NULL, return the root window's colormap,
  * also, guard against XErrors
  */
+static
 GdkColormap*
 gdk_window_get_colormap (GdkWindow *window)
 {
@@ -707,6 +716,7 @@
 /* instead of bailing on NULL, return the root window's visual, also
  * guard against XErrors
  */
+static
 GdkVisual*
 gdk_window_get_visual (GdkWindow *window)
 {
@@ -875,6 +885,7 @@
 			     x, y, width, height, cmap);
 }
 
+static
 void
 gdk_pixbuf_copy_from_image (GdkPixbuf   *pixbuf,
 			    guint	 dest_x,
@@ -908,6 +919,7 @@
 		      cmap);
 }
 
+static
 GdkPixbuf*
 gdk_pixbuf_from_image (GdkImage    *image,
 		       guint        x,
diff -ru gnome-core-1.2.4/applets/fish/Makefile.am gnome-core-1.2.4-patched/applets/fish/Makefile.am
--- gnome-core-1.2.4/applets/fish/Makefile.am	Tue Feb 29 09:49:13 2000
+++ gnome-core-1.2.4-patched/applets/fish/Makefile.am	Thu Jan 18 16:41:49 2001
@@ -15,6 +15,7 @@
 #endif
 
 libfish_applet_la_SOURCES = fish.c
+libfish_applet_la_LDFLAGS = -module
 
 #fish_applet_SOURCES = fish.c
 
diff -ru gnome-core-1.2.4/applets/fish/Makefile.in gnome-core-1.2.4-patched/applets/fish/Makefile.in
--- gnome-core-1.2.4/applets/fish/Makefile.in	Wed Nov 15 20:20:40 2000
+++ gnome-core-1.2.4-patched/applets/fish/Makefile.in	Thu Jan 18 16:40:39 2001
@@ -187,7 +187,7 @@
 
 DEFS = @DEFS@ -I. -I$(srcdir) -I../..
 LIBS = @LIBS@
-libfish_applet_la_LDFLAGS = 
+libfish_applet_la_LDFLAGS = -module
 libfish_applet_la_LIBADD = 
 libfish_applet_la_OBJECTS =  fish.lo
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
diff -ru gnome-core-1.2.4/applets/gen_util/Makefile.am gnome-core-1.2.4-patched/applets/gen_util/Makefile.am
--- gnome-core-1.2.4/applets/gen_util/Makefile.am	Tue Sep  5 20:35:18 2000
+++ gnome-core-1.2.4-patched/applets/gen_util/Makefile.am	Thu Jan 18 16:44:28 2001
@@ -19,6 +19,7 @@
 	printer.h	\
 	clock.c		\
 	clock.h
+libgen_util_applet_la_LDFLAGS = -module
 
 PRINTER_BITMAPS = print.xpm
 MAILCHECK_BITMAPS = email.png email-e.png tux-anim.png email-eclouds.png \
diff -ru gnome-core-1.2.4/applets/gen_util/Makefile.in gnome-core-1.2.4-patched/applets/gen_util/Makefile.in
--- gnome-core-1.2.4/applets/gen_util/Makefile.in	Wed Nov 15 20:20:48 2000
+++ gnome-core-1.2.4-patched/applets/gen_util/Makefile.in	Thu Jan 18 16:44:17 2001
@@ -181,7 +181,7 @@
 
 DEFS = @DEFS@ -I. -I$(srcdir) -I../..
 LIBS = @LIBS@
-libgen_util_applet_la_LDFLAGS = 
+libgen_util_applet_la_LDFLAGS = -module
 libgen_util_applet_la_LIBADD = 
 libgen_util_applet_la_OBJECTS =  main.lo mailcheck.lo popcheck.lo \
 printer.lo clock.lo
