diff -ru gnome-libs-1.2.8/configure.in gnome-libs-1.2.8-patched/configure.in
--- gnome-libs-1.2.8/configure.in	Tue Oct 24 01:15:03 2000
+++ gnome-libs-1.2.8-patched/configure.in	Mon Jan 15 07:41:35 2001
@@ -256,7 +256,7 @@
     AC_CHECK_LIB(tiff34, TIFFFlushData,
       AC_CHECK_HEADER(tiffio.h,
         TIFF_LIBS="-ltiff34 $JPEG_LIBS $Z_LIBS"; AC_DEFINE(HAVE_LIBTIFF)),,
-$JPEG_LIBS $Z_LIBS -lm), $JPEG_LIBS $Z_LIBS -lm), -lm)
+$JPEG_LIBS $Z_LIBS), $JPEG_LIBS $Z_LIBS), )
 
 dnl Test for libgif
 AC_CHECK_LIB(gif, DGifOpenFileName,
@@ -272,7 +272,7 @@
 AC_CHECK_LIB(png, png_read_info,
   AC_CHECK_HEADER(png.h,
     png_ok=yes,
-    png_ok=no),,$Z_LIBS -lm)
+    png_ok=no),,$Z_LIBS)
 if test "$png_ok" = yes; then
   AC_MSG_CHECKING([for png_structp in png.h])
   AC_TRY_COMPILE([#include <png.h>],
diff -ru gnome-libs-1.2.8/libgnomeui/gnome-canvas-load.c gnome-libs-1.2.8-patched/libgnomeui/gnome-canvas-load.c
--- gnome-libs-1.2.8/libgnomeui/gnome-canvas-load.c	Tue Aug 24 02:24:48 1999
+++ gnome-libs-1.2.8-patched/libgnomeui/gnome-canvas-load.c	Mon Jan 15 10:14:23 2001
@@ -12,7 +12,9 @@
  */
 #include <config.h>
 #include <gdk_imlib.h>
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
 #include "gnome-canvas.h"
 #include "gnome-canvas-load.h"
 
diff -ru gnome-libs-1.2.8/zvt/gnome-pty-helper.c gnome-libs-1.2.8-patched/zvt/gnome-pty-helper.c
--- gnome-libs-1.2.8/zvt/gnome-pty-helper.c	Sat Mar 18 08:57:13 2000
+++ gnome-libs-1.2.8-patched/zvt/gnome-pty-helper.c	Mon Jan 15 10:32:35 2001
@@ -45,7 +45,9 @@
 #include <fcntl.h>
 #include <termios.h>
 #include <errno.h>
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
 #include <termios.h>
 #include <pwd.h>
 #include <stdlib.h>
diff -ru gnome-libs-1.2.8/gnome-config.in gnome-libs-1.2.8-patched/gnome-config.in
--- gnome-libs-1.2.8/gnome-config.in	Thu Sep 28 01:59:01 2000
+++ gnome-libs-1.2.8-patched/gnome-config.in	Tue Jan 16 17:41:45 2001
@@ -177,7 +177,7 @@
   	the_flags="$the_flags @GNOME_INCLUDEDIR@ @GTK_CFLAGS@"
 	;;
     gnomeui)
-	the_libs="$the_libs @GNOME_LIBDIR@ @GNOMEUI_LIBS@"
+	the_libs="$the_libs @GNOME_LIBDIR@ @GNOMEUI_LIBS@ @INTLLIBS@"
 	the_flags="$the_flags @GNOME_INCLUDEDIR@ @GTK_CFLAGS@"
 	;;
     gtkxmhtml)
