diff -ru gnome-applets-1.2.4/Makefile.am gnome-applets-1.2.4-patched/Makefile.am
--- gnome-applets-1.2.4/Makefile.am	Thu Aug  3 05:11:40 2000
+++ gnome-applets-1.2.4-patched/Makefile.am	Tue Jan 23 18:58:55 2001
@@ -69,7 +69,6 @@
 	gkb-new		\
 	jbc		\
 	life		\
-	mini-commander	\
 	quicklaunch	\
 	tickastat	\
 	webcontrol	\
diff -ru gnome-applets-1.2.4/Makefile.in gnome-applets-1.2.4-patched/Makefile.in
--- gnome-applets-1.2.4/Makefile.in	Wed Nov 15 21:04:53 2000
+++ gnome-applets-1.2.4-patched/Makefile.in	Tue Jan 23 18:58:45 2001
@@ -181,7 +181,7 @@
 
 @APPLET_MODEMLIGHTS_TRUE@modemlights_SUBDIRS = $(modemlights_applets)
 
-always_built_SUBDIRS =  	another_clock		asclock			battery			charpick		clockmail		drivemount		fifteen			geyes			gkb-new			jbc			life			mini-commander		quicklaunch		tickastat		webcontrol		whereami		odometer
+always_built_SUBDIRS =  	another_clock		asclock			battery			charpick		clockmail		drivemount		fifteen			geyes			gkb-new			jbc			life					quicklaunch		tickastat		webcontrol		whereami		odometer
 
 
 SUBDIRS =  	intl po macros 			$(always_built_SUBDIRS)		$(gtop_SUBDIRS)			$(ghttp_SUBDIRS)		$(ghttp_xml_SUBDIRS)		$(cdplayer_SUBDIRS)		$(mixer_SUBDIRS)		$(modemlights_SUBDIRS)		$(esd_SUBDIRS)			$(wordexp_SUBDIRS)      	$(xml_SUBDIRS)
diff -ru gnome-applets-1.2.4/jbc/jbc-applet.c gnome-applets-1.2.4-patched/jbc/jbc-applet.c
--- gnome-applets-1.2.4/jbc/jbc-applet.c	Thu May  4 14:23:59 2000
+++ gnome-applets-1.2.4-patched/jbc/jbc-applet.c	Tue Jan 23 14:57:08 2001
@@ -72,7 +72,8 @@
 do_flicker ()
 {
   thetime = time (NULL);
-  localtime_r (&thetime, &atime);
+  /* localtime_r (&thetime, &atime); */
+  atime = * localtime(&thetime);
 
   d[0] = atime.tm_hour / 10;
   d[1] = atime.tm_hour % 10;
diff -ru gnome-applets-1.2.4/mini-commander/src/exec.c gnome-applets-1.2.4-patched/mini-commander/src/exec.c
--- gnome-applets-1.2.4/mini-commander/src/exec.c	Tue Feb  1 20:11:39 2000
+++ gnome-applets-1.2.4-patched/mini-commander/src/exec.c	Tue Jan 23 18:40:54 2001
@@ -34,7 +34,12 @@
 
 /*in case SA_NOMASK doesn't exist, use SA_NODEFER*/
 #ifndef SA_NOMASK
+#ifndef SA_NODEFER
+#warning Neither SA_NOMASK nor SA_NODEFER are declared. Signal handling may be affected.
+#define SA_NOMASK (0)
+#else
 #define SA_NOMASK SA_NODEFER
+#endif
 #endif
 
 /* 
