diff -ru gnome-utils-1.2.1/Makefile.in gnome-utils-1.2.1-patched/Makefile.in
--- gnome-utils-1.2.1/Makefile.in	Tue Jul 11 00:52:31 2000
+++ gnome-utils-1.2.1-patched/Makefile.in	Tue Mar  6 20:42:42 2001
@@ -167,7 +167,7 @@
 
 @FALSE_TRUE@extras = gnome-find
 
-SUBDIRS = po intl macros idl $(gfloppy) logview gcharmap 	gcalc gsearchtool $(gdiskfree) $(gstripchart) gtt gdictsrc 	$(guile_DIRS) $(objc_DIRS) mini-utils splash gdialog $(extras)
+SUBDIRS = po intl macros idl $(gfloppy) logview gcharmap 	gcalc gsearchtool $(gdiskfree) $(gstripchart) gtt gdictsrc 	$(guile_DIRS) $(objc_DIRS) mini-utils splash $(extras)
 
 
 EXTRA_DIST = gnome-utils.spec.in
diff -ru gnome-utils-1.2.1/logview/log_repaint.c gnome-utils-1.2.1-patched/logview/log_repaint.c
--- gnome-utils-1.2.1/logview/log_repaint.c	Tue Jul 11 00:49:14 2000
+++ gnome-utils-1.2.1-patched/logview/log_repaint.c	Tue Mar  6 20:30:32 2001
@@ -23,7 +23,9 @@
 #include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
 #include <config.h>
 #include <string.h>
 #include <gnome.h>
diff -ru gnome-utils-1.2.1/logview/logrtns.c gnome-utils-1.2.1-patched/logview/logrtns.c
--- gnome-utils-1.2.1/logview/logrtns.c	Thu Jun 15 20:17:52 2000
+++ gnome-utils-1.2.1-patched/logview/logrtns.c	Tue Mar  6 20:38:01 2001
@@ -698,7 +698,7 @@
       if (IsLeapYear (curmark->year - lastyear + thisyear))
 	 curmark->time += 24 * 60 * 60;		/*  Add one day */
 
-#ifdef __NetBSD__
+#if defined( __NetBSD__ ) || defined( __APPLE__ )
       curmark->time += correction - tmptm->tm_gmtoff;
 #else
       curmark->time += correction - timezone;
@@ -724,7 +724,7 @@
    if (IsLeapYear (thisyear))
       log->lstats.enddate += 24 * 60 * 60;	/*  Add one day */
 
-#ifdef __NetBSD__
+#if defined( __NetBSD__ ) || defined( __APPLE__ )
    log->lstats.enddate += correction - tmptm->tm_gmtoff;
 #else
    log->lstats.enddate += correction - timezone;
@@ -735,7 +735,7 @@
    if (IsLeapYear (thisyear - lastyear))
       log->lstats.startdate += 24 * 60 * 60;	/*  Add one day */
 
-#ifdef __NetBSD__
+#if defined( __NetBSD__ ) || defined( __APPLE__ )
    log->lstats.startdate += correction - tmptm->tm_gmtoff;
 #else
    log->lstats.startdate += correction - timezone;
diff -ru gnome-utils-1.2.1/logview/logrtns.h gnome-utils-1.2.1-patched/logview/logrtns.h
--- gnome-utils-1.2.1/logview/logrtns.h	Thu Apr 27 05:32:02 2000
+++ gnome-utils-1.2.1-patched/logview/logrtns.h	Tue Mar  6 20:30:55 2001
@@ -24,7 +24,9 @@
 #include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
 #include <string.h>
 #include "logview.h"
 
