--- ./iceMC-0.1.1-beta/Makefile.orig	Mon Mar 12 05:48:39 2001
+++ ./iceMC-0.1.1-beta/Makefile	Mon Mar 12 05:48:58 2001
@@ -7,14 +7,14 @@
 
 ####### Compiler, tools and options
 
-CC	=	gcc
-CXX	=	g++
+CC	=	cc
+CXX	=	c++
 CFLAGS	=	-Wall -pipe
-CXXFLAGS=	-Wall -pipe
+CXXFLAGS=	-Wall -pipe 
 INCPATH	=	-I$(QTDIR)/include
-LINK	=	g++
+LINK	=	c++
 LFLAGS	=	
-LIBS	=	$(SUBLIBS) -L$(QTDIR)/lib -L/usr/X11R6/lib -lqt -lXext -lX11 -lm
+LIBS	=	$(SUBLIBS) -L$(QTDIR)/lib -L/usr/X11R6/lib -lqt -lXext -lSM -lICE -lXft -lXrender -lX11
 MOC	=	$(QTDIR)/bin/moc
 UIC	=	$(QTDIR)/bin/uic
 
