diff -ru icewm-1.0.6/src/yapp.cc icewm-1.0.6-patched/src/yapp.cc
--- icewm-1.0.6/src/yapp.cc	Sun Jan 14 22:49:41 2001
+++ icewm-1.0.6-patched/src/yapp.cc	Thu Jan 18 11:29:18 2001
@@ -1178,7 +1178,7 @@
     sigprocmask(SIG_BLOCK, &signalMask, NULL);
 
     struct sigaction sa;
-    sa.sa_handler = sig_handler;
+    sa.sa_handler = ( void (*)() )sig_handler;
     sigemptyset(&sa.sa_mask);
     sa.sa_flags = 0;
     sigaction(sig, &sa, NULL);
