--- less-358/configure.in	Sun Jul  9 02:26:43 2000
+++ less-358-patched/configure.in	Fri Dec 22 22:11:50 2000
@@ -114,6 +114,17 @@
   fi
 fi
 
+dnl -- Try curses in the system library.
+if test "x$TERMLIBS" = x; then
+  TERMLIBS="-L."
+  SAVE_LIBS=$LIBS
+  LIBS="$LIBS $TERMLIBS"
+  AC_TRY_LINK(, [tgetent(0,0); tgetflag(0); tgetnum(0); tgetstr(0,0);],
+    [termok=yes], [termok=no])
+  LIBS=$SAVE_LIBS
+  if test $termok = no; then TERMLIBS=""; fi
+fi
+
 if test "x$TERMLIBS" = x; then
   AC_MSG_RESULT(Cannot find terminal libraries - configure failed)
   exit 1
