--- lesstif-0.91.8/lib/Xm/BaseClass.c	Thu Feb  1 10:12:48 2001
+++ lesstif-0.91.8/lib/Xm/BaseClass.c	Thu Feb  1 10:13:26 2001
@@ -650,7 +650,7 @@
 
     while ((wc != NULL) && (wc != (WidgetClass)vendorShellWidgetClass))
     {
-	Depth++;
+	;Depth++;
 	wc = wc->core_class.superclass;
     }
 
@@ -688,7 +688,7 @@
 
     while ((wc != NULL) && (wc != (WidgetClass)rectObjClass))
     {
-	Depth++;
+	;Depth++;
 	wc = wc->core_class.superclass;
     }
 
@@ -728,7 +728,7 @@
 
     while ((wc != NULL) && (wc != (WidgetClass)rectObjClass))
     {
-	Depth++;
+	;Depth++;
 	wc = wc->core_class.superclass;
     }
 
--- lesstif-0.91.8/clients/Motif-1.2/mwm/mwm.c.orig	Thu Feb  1 09:32:13 2001
+++ lesstif-0.91.8/clients/Motif-1.2/mwm/mwm.c	Thu Feb  1 09:34:10 2001
@@ -115,9 +115,9 @@
 sig_restart(int sig)
 {
     MWM_Done(1, *g_argv);
-#if RETSIGTYPE != void
+/*#if RETSIGTYPE != void*/
     return (RETSIGTYPE) 0;
-#endif
+/*#endif*/
 }
 
 /*
@@ -127,9 +127,9 @@
 sig_done(int nonsense)
 {
     MWM_Done(0, NULL);
-#if RETSIGTYPE != void
+/*#if RETSIGTYPE != void*/
     return (RETSIGTYPE) 0;
-#endif
+/*#endif*/
 }
 
 /*
@@ -141,9 +141,9 @@
     alarmed = True;
     signal(SIGALRM, sig_alarm);
 
-#if RETSIGTYPE != void
+/*#if RETSIGTYPE != void*/
     return (RETSIGTYPE) 0;
-#endif
+/*#endif*/
 }
 
 
--- lesstif-0.91.8/clients/Motif-1.2/mwm/screens.c.orig	Thu Feb  1 10:10:23 2001
+++ lesstif-0.91.8/clients/Motif-1.2/mwm/screens.c	Thu Feb  1 10:11:38 2001
@@ -84,8 +84,8 @@
 	XCreateGC(dpy, scr->root_win, gcm, &gcv);
 
     {
-        Pixmap stipple = XCreateBitmapFromData(dpy, scr->root_win, stipple_bits,
-                                     stipple_width, stipple_height);
+        Pixmap stipple = XCreateBitmapFromData(dpy, scr->root_win, Stipple_bits,
+                                     Stipple_width, Stipple_height);
         if (stipple) {
             XSetFillStyle(dpy, scr->components[MWM_MENU].grayed_GC,
                           FillOpaqueStippled);
