diff -ru tiff-v3.5.5/config.site tiff-v3.5.5-patched/config.site
--- tiff-v3.5.5/config.site	Mon Sep 27 00:03:46 1999
+++ tiff-v3.5.5-patched/config.site	Wed Feb 28 19:56:47 2001
@@ -37,12 +37,15 @@
 # interactive prompting done by the configure script.
 #
 
+PREFIX=@PREFIX@
+STOWDIR=@STOWDIR@
+
 #
 # Package controls.
 #
-#DSO="auto"			# auto|IRIX|IRIX52 enable DSO support for system
-#JPEG="no"			# yes|no configure JPEG support (see below)
-#ZIP="no"			# yes|no configure Deflate support (see below)
+DSO="auto"			# auto|IRIX|IRIX52 enable DSO support for system
+JPEG="yes"			# yes|no configure JPEG support (see below)
+ZIP="yes"			# yes|no configure Deflate support (see below)
 #LIBGL="auto"			# yes|no|auto configure IRIS GL-based tools
 #LIBIMAGE="auto"		# yes|no|auto configure SGI RGB image tools
 #HTML="no"			# yes|no install HTML documentation
@@ -50,10 +53,10 @@
 #
 # Directory parameters.
 #
-#DIR_BIN="/usr/local/bin"		# directory for tools
-#DIR_LIB="/usr/local/lib"		# directory for libraries
-#DIR_INC="/usr/local/include"		# directory for include files
-#DIR_MAN="/usr/local/man"		# directory for manual pages
+DIR_BIN="$STOWDIR/bin"		# directory for tools
+DIR_LIB="$STOWDIR/lib"		# directory for libraries
+DIR_INC="$STOWDIR/include"		# directory for include files
+DIR_MAN="$STOWDIR/man"		# directory for manual pages
 #DIR_HTML="/var/httpd/htdocs/tiff"	# directory for HTML documentation
 
 #
@@ -95,7 +98,9 @@
 # which you can find on the Internet at ftp.uu.net:/graphics/jpeg/.
 #
 #DIRS_LIBINC="$DIRS_LIBINC ../src/jpeg-6b" # directory for jpeg include files
+DIRS_LIBINC="$DIRS_LIBINC $PREFIX/include"
 #DIR_JPEGLIB=../src/jpeg-6b		# directory for libjpeg
+DIR_JPEGLIB="$PREFIX/lib"
 
 #
 # Deflate-specific parameters; used when Deflate support is enabled (see above).
@@ -110,7 +115,9 @@
 #     Look for it also at ftp://ftp.uu.net/pub/archiving/zip/zlib.
 #
 #DIRS_LIBINC="$DIRS_LIBINC ../zlib"	# directory for zlib include files
+DIRS_LIBINC="$DIRS_LIBINC $PREFIX/include"	# directory for zlib include files
 #DIR_GZLIB="../zlib"			# directory for libz
+DIR_GZLIB="$PREFIX/lib"			# directory for libz
 
 #
 # Miscellaneous parameters.
@@ -133,7 +140,7 @@
 #ENVOPTS="-Aa"				# options for getting ANSI C
 #GCOPTS="-g"				# options to pass C compiler
 #LIBPORT='${PORT}/libport.a'		# library with emulation code
-#MACHDEPLIBS=""				# extra libraries for linking
+MACHDEPLIBS=""				# extra libraries for linking
 #PORTFUNCS=""				# non-standard functions to emulate
 #RANLIB=":"				# pathname of suitable ranlib program
 #DSOSUF="so"				# DSO filename suffix
diff -ru tiff-v3.5.5/configure tiff-v3.5.5-patched/configure
--- tiff-v3.5.5/configure	Mon Mar 20 19:20:42 2000
+++ tiff-v3.5.5-patched/configure	Wed Feb 28 20:17:06 2001
@@ -1169,6 +1169,13 @@
 	DSOOPTS='-shared'
 	DSO=OSF
 	;;
+    *-darwin*)
+	DSOSUF=dylib
+	DSOLD=cc
+	DSOOPTS='-dynamiclib -undefined suppress'
+	LIBCOPTS='-fno-common'
+	DSO=DARWIN
+	;;
     esac
 fi
 if [ "$DSO" != no ]; then
diff -ru tiff-v3.5.5/libtiff/Makefile.in tiff-v3.5.5-patched/libtiff/Makefile.in
--- tiff-v3.5.5/libtiff/Makefile.in	Sun Nov 28 21:15:36 1999
+++ tiff-v3.5.5-patched/libtiff/Makefile.in	Wed Feb 28 20:00:47 2001
@@ -215,6 +215,10 @@
 # OSF/1 3.2 shared lib rule
 OSFdso: ${OBJS}
 	${LD} -o libtiff.@DSOSUF@ -shared -error_unresolved ${OBJS} @LIBJPEG@ @LIBGZ@ -lc -lm
+# Darwin 1.2 shared lib rule
+DARWINdso: ${OBJS}
+	${CC} -dynamiclib -undefined suppress -install_name @DIR_LIB@/libtiff.@DSOSUF@ -o libtiff.@DSOSUF@ ${OBJS} @LIBJPEG@ @LIBGZ@
+	touch $@
 
 
 ${OBJS}: ${SRCDIR}/tiffio.h ${SRCDIR}/tiff.h ${SRCDIR}/tif_dir.h
