diff -ru lynx2-8-3/WWW/Library/Implementation/HTTP.c lynx2-8-3-patched/WWW/Library/Implementation/HTTP.c
--- lynx2-8-3/WWW/Library/Implementation/HTTP.c	Sat Feb 24 14:00:09 2001
+++ lynx2-8-3-patched/WWW/Library/Implementation/HTTP.c	Sat Feb 24 14:40:27 2001
@@ -111,7 +111,8 @@
     char rand_file[256];
     time_t t;
     pid_t pid;
-    long l,seed;
+    long seed;
+    int i;
 
     t=time(NULL);
     pid=getpid();
@@ -127,11 +128,11 @@
     RAND_seed((unsigned char *)&pid,sizeof(pid_t));
     /* Initialize system's random number generator */
     RAND_bytes((unsigned char *)&seed,sizeof(long));
-    srand48(seed);
+    srand((unsigned int)seed);
     while(RAND_status()==0) {
       /* Repeatedly seed the PRNG using the system's random number generator until it has been seeded with enough data */
-      l=lrand48();
-      RAND_seed((unsigned char *)&l,sizeof(long));
+      i=rand();
+      RAND_seed((unsigned char *)&i,sizeof(int));
     }
     if(rand_file!=NULL) {
       /* Write a rand_file */
