--- mysql-3.23.36/sql/mysqld.cc	Tue Mar 27 02:11:02 2001
+++ mysql-3.23.36/sql/mysqld.cc	Fri Mar 30 21:20:50 2001
@@ -1237,11 +1237,7 @@
   sigprocmask(SIG_SETMASK,&sa.sa_mask,NULL);
   if (!(test_flags & TEST_NO_STACKTRACE))
   {
-#ifdef HAVE_DARWIN_THREADS
-    sa.sa_handler=( void (*)() ) handle_segfault;
-#else
     sa.sa_handler=handle_segfault;
-#endif
     sigaction(SIGSEGV, &sa, NULL);
 #ifdef SIGBUS
     sigaction(SIGBUS, &sa, NULL);
