diff -ru openssl-0.9.6/Configure openssl-0.9.6-patched/Configure
--- openssl-0.9.6/Configure	Sun Sep 24 17:27:37 2000
+++ openssl-0.9.6-patched/Configure	Tue Mar 27 21:04:12 2001
@@ -404,6 +404,8 @@
 
 ##### MacOS X (a.k.a. Rhapsody) setup
 "rhapsody-ppc-cc","cc:-O3 -DB_ENDIAN::(unknown)::BN_LLONG RC4_CHAR RC4_CHUNK DES_UNROLL BF_PTR:::",
+"Rhapsody",	"cc:-O3 -fomit-frame-pointer -Wall::(unknown)::::",
+"Darwin",	"cc:-O3 -fomit-frame-pointer -Wall::(unknown):::::::::::::darwin-shared:-fno-common",
 
 ##### Sony NEWS-OS 4.x
 "newsos4-gcc","gcc:-O -DB_ENDIAN -DNEWS4::(unknown):-lmld -liberty:BN_LLONG RC4_CHAR RC4_CHUNK DES_PTR DES_RISC1 DES_UNROLL BF_PTR::::",
@@ -715,8 +717,8 @@
 	$cflags = "$shared_cflag $cflags";
 	if (!$no_shared)
 		{
-		$shared_mark1 = ".shlib-clean.";
-		$shared_mark2 = ".shlib.";
+		$shared_mark1 = ".shlib.";
+		$shared_mark2 = "";
 		}
 	}
 
diff -ru openssl-0.9.6/Makefile.org openssl-0.9.6-patched/Makefile.org
--- openssl-0.9.6/Makefile.org	Thu Sep 21 11:23:13 2000
+++ openssl-0.9.6-patched/Makefile.org	Tue Mar 27 21:31:23 2001
@@ -170,7 +170,7 @@
 MAKEFILE= Makefile.ssl
 MAKE=     make -f Makefile.ssl
 
-MANDIR=$(OPENSSLDIR)/man
+MANDIR=$(INSTALLTOP)/man
 MAN1=1
 MAN3=3
 SHELL=/bin/sh
@@ -292,6 +292,21 @@
 	ln -s lib$$i.so.${SHLIB_MAJOR}.${SHLIB_MINOR} \
 		lib$$i.so.${SHLIB_MAJOR}; \
 	ln -s lib$$i.so.${SHLIB_MAJOR} lib$$i.so; \
+	done
+
+darwin-shared:
+	libs='${SHLIBDEPS}'; for i in ${SHLIBDIRS}; do \
+	rm -f lib$$i.dylib \
+		lib$$i.${SHLIB_MAJOR}.${SHLIB_MINOR}.dylib; \
+	( set -x; ${CC} -dynamiclib -o lib$$i.${SHLIB_MAJOR}.${SHLIB_MINOR}.dylib \
+		-install_name ${INSTALLTOP}/lib/lib$$i.${SHLIB_MAJOR}.${SHLIB_MINOR}.dylib \
+		-compatibility_version ${SHLIB_MAJOR}.${SHLIB_MINOR} \
+		-current_version ${SHLIB_MAJOR}.${SHLIB_MINOR} \
+		-all_load lib$$i.a \
+		$$libs ${EX_LIBS} ) || exit 1; \
+	libs="$$libs -L. -l$$i"; \
+	( set -x; \
+		ln -s lib$$i.${SHLIB_MAJOR}.${SHLIB_MINOR}.dylib lib$$i.dylib ); \
 	done
 
 Makefile.ssl: Makefile.org
diff -ru openssl-0.9.6/config openssl-0.9.6-patched/config
--- openssl-0.9.6/config	Thu Sep 21 11:23:14 2000
+++ openssl-0.9.6-patched/config	Tue Mar 27 19:59:33 2001
@@ -179,6 +179,18 @@
 	echo "${MACHINE}-whatever-openbsd"; exit 0
 	;;
 
+    Rhapsody:*)
+	case ${MACHINE} in
+	    Power*) MACH="powerpc"    ;;
+	    *     ) MACH="${MACHINE}" ;;
+	esac
+	echo "${MACH}-apple-rhapsody${RELEASE}"; exit 0
+	;;
+
+    Darwin:*|"Mac OS":*)
+	echo "`uname -p`-apple-darwin${RELEASE}"; exit 0
+	;;
+
     OSF1:*:*:*alpha*)
 	echo "${MACHINE}-dec-osf"; exit 0
 	;;
@@ -461,6 +473,8 @@
   alpha*-*-openbsd) OUT="OpenBSD-alpha" ;;
   pmax*-*-openbsd) OUT="OpenBSD-mips" ;;
   *-*-openbsd) OUT="OpenBSD" ;;
+  *-*-rhapsody*) OUT="Rhapsody" ;;
+  *-*-darwin*) OUT="Darwin" ;;
   *86*-*-bsdi4) OUT="bsdi-elf-gcc" ;;
   *-*-osf) OUT="alpha-cc" ;;
   *-*-unixware7) OUT="unixware-7" ;;
diff -ru openssl-0.9.6/crypto/opensslconf.h.in openssl-0.9.6-patched/crypto/opensslconf.h.in
--- openssl-0.9.6/crypto/opensslconf.h.in	Fri Feb 18 10:06:55 2000
+++ openssl-0.9.6-patched/crypto/opensslconf.h.in	Tue Mar 27 19:59:33 2001
@@ -144,7 +144,7 @@
 #  define DES_PTR
 #  define DES_RISC2
 #  define DES_UNROLL
-#elif defined( i386 )		/* x86 boxes, should be gcc */
+#elif defined( i386 ) || defined( __i386__ )	/* x86 boxes, should be gcc */
 #  define DES_PTR
 #  define DES_RISC1
 #  define DES_UNROLL
