diff -ru openssl-0.9.5a/Configure openssl-0.9.5a-patched/Configure
--- openssl-0.9.5a/Configure	Mon Mar 27 23:28:10 2000
+++ openssl-0.9.5a-patched/Configure	Thu Feb 22 14:19:28 2001
@@ -372,6 +372,8 @@
 
 ##### MacOS X (a.k.a. Rhapsody) setup
 "rhapsody-ppc-cc","cc:-O3 -DB_ENDIAN::(unknown)::BN_LLONG RC4_CHAR RC4_CHUNK DES_UNROLL BF_PTR:::",
+"Rhapsody",	"cc:-O3 -fomit-frame-pointer -Wall::(unknown)::::",
+"Darwin",	"cc:-O3 -fomit-frame-pointer -Wall::(unknown)::::",
 
 );
 
diff -ru openssl-0.9.5a/Makefile.org openssl-0.9.5a-patched/Makefile.org
--- openssl-0.9.5a/Makefile.org	Thu Mar 16 20:46:20 2000
+++ openssl-0.9.5a-patched/Makefile.org	Thu Feb 22 14:19:28 2001
@@ -156,7 +156,7 @@
 MAKEFILE= Makefile.ssl
 MAKE=     make -f Makefile.ssl
 
-MANDIR=$(OPENSSLDIR)/man
+MANDIR=$(INSTALLTOP)/man
 MAN1=1
 MAN3=3
 SHELL=/bin/sh
diff -ru openssl-0.9.5a/config openssl-0.9.5a-patched/config
--- openssl-0.9.5a/config	Tue Mar 14 00:52:44 2000
+++ openssl-0.9.5a-patched/config	Thu Feb 22 14:19:28 2001
@@ -175,6 +175,18 @@
 	echo "${MACHINE}-whatever-openbsd"; exit 0
 	;;
 
+    Rhapsody:*)
+	case ${MACHINE} in
+	    Power*) MACH="powerpc"    ;;
+	    *     ) MACH="${MACHINE}" ;;
+	esac
+	echo "${MACH}-apple-rhapsody${RELEASE}"; exit 0
+	;;
+
+    Darwin:*|"Mac OS":*)
+	echo "`uname -p`-apple-darwin${RELEASE}"; exit 0
+	;;
+
     OSF1:*:*:*alpha*)
 	echo "${MACHINE}-dec-osf"; exit 0
 	;;
@@ -453,6 +465,8 @@
   alpha*-*-openbsd) OUT="OpenBSD-alpha" ;;
   pmax*-*-openbsd) OUT="OpenBSD-mips" ;;
   *-*-openbsd) OUT="OpenBSD" ;;
+  *-*-rhapsody*) OUT="Rhapsody" ;;
+  *-*-darwin*) OUT="Darwin" ;;
   *86*-*-bsdi4) OUT="bsdi-elf-gcc" ;;
   *-*-osf) OUT="alpha-cc" ;;
   *-*-unixware7) OUT="unixware-7" ;;
diff -ru openssl-0.9.5a/crypto/opensslconf.h.in openssl-0.9.5a-patched/crypto/opensslconf.h.in
--- openssl-0.9.5a/crypto/opensslconf.h.in	Fri Feb 18 10:06:55 2000
+++ openssl-0.9.5a-patched/crypto/opensslconf.h.in	Thu Feb 22 14:19:28 2001
@@ -144,7 +144,7 @@
 #  define DES_PTR
 #  define DES_RISC2
 #  define DES_UNROLL
-#elif defined( i386 )		/* x86 boxes, should be gcc */
+#elif defined( i386 ) || defined( __i386__ )	/* x86 boxes, should be gcc */
 #  define DES_PTR
 #  define DES_RISC1
 #  define DES_UNROLL
