diff -ru ORBit-0.5.6/orbit-config.in ORBit-0.5.6-patched/orbit-config.in
--- ORBit-0.5.6/orbit-config.in	Fri Oct  8 06:49:07 1999
+++ ORBit-0.5.6-patched/orbit-config.in	Sat Jan 27 09:45:41 2001
@@ -87,7 +87,7 @@
        	;;
 
     client|server)
-	the_libs="$the_libs -L$libdir -lORBit -lIIOP -lORBitutil `glib-config --libs` @LIBS@ -lm"
+	the_libs="$the_libs -L$libdir -lORBit -lIIOP -lORBitutil `glib-config --libs` @LIBS@"
 	the_flags="$the_flags `glib-config --cflags` -I$includedir "
 	;;
 
