diff -ru pine4.33/build pine4.33-patched/build
--- pine4.33/build	Thu Jan  4 19:13:11 2001
+++ pine4.33-patched/build	Fri Feb 23 15:30:33 2001
@@ -303,7 +303,7 @@
 
 # Try to automatically include SSL. To prevent SSL from being automatically
 # included use "build NOSSL target".
-        if [ -d /usr/local/ssl -a "$SAFLAGS" != "1" -a "$NOSSL" != "1" ]
+        if [ "$SAFLAGS" != "1" -a "$NOSSL" != "1" ]
 	then
 	  SSL="'SPECIALAUTHENTICATORS=ssl'"
 	  echo "Including SSL functionality"
