--- pwm-20010110/system.mk	Thu Dec 28 07:08:05 2000
+++ pwm-20010110-new/system.mk	Thu Feb  1 13:13:34 2001
@@ -40,7 +40,7 @@
 ## C compiler
 ##
 
-CC=gcc
+CC=cc
 
 # The POSIX_SOURCE, XOPEN_SOURCE and WARN options should not be necessary,
 # they're mainly for development use. So, if they cause trouble (not
@@ -62,12 +62,12 @@
 
 # Same as '-Wall -pedantic-errors' without '-Wunused' as callbacks often
 # have unused variables.
-WARN=	-W -Wimplicit -Wreturn-type -Wswitch -Wcomment \
+#WARN=	-W -Wimplicit -Wreturn-type -Wswitch -Wcomment \
 	-Wtrigraphs -Wformat -Wchar-subscripts \
 	-Wparentheses -pedantic-errors -Wuninitialized
 
 
-CFLAGS=-g -O2 $(WARN) $(DEFINES) $(INCLUDES) $(EXTRA_INCLUDES)
+CFLAGS=-g -O2 -traditional-cpp $(WARN) $(DEFINES) $(INCLUDES) $(EXTRA_INCLUDES)
 LDFLAGS=-g $(LIBS) $(EXTRA_LIBS)
 
 
--- pwm-20010110/Makefile	Wed Jan 10 11:15:28 2001
+++ pwm-20010110-new/Makefile	Mon Jan 15 09:20:36 2001
@@ -8,7 +8,7 @@
 ######################################
 
 SUBDIRS=libtu
-LIBS += -L./libtu -ltu -lm $(X11_LIBS) -lX11
+LIBS += -L./libtu -ltu $(X11_LIBS) -lX11
 INCLUDES += -I./libtu/include $(X11_INCLUDES)
 DEFINES += -DETCDIR=\"$(ETCDIR)\"
 CFLAGS += $(XOPEN_SOURCE)
