diff -ruN qt-2.3.0/configs/darwin-c++-shared qt-2.3.0-patched/configs/darwin-c++-shared
--- qt-2.3.0/configs/darwin-c++-shared	Thu Jan  1 01:00:00 1970
+++ qt-2.3.0-patched/configs/darwin-c++-shared	Fri Mar 16 11:24:37 2001
@@ -0,0 +1,98 @@
+# Compiling
+INTERFACE_DECL_PATH 	= .
+SYSCONF_CXX		= c++
+SYSCONF_CC		= cc
+DASHCROSS		= 
+
+# Compiling with support libraries
+SYSCONF_CXXFLAGS_X11	= -I/usr/X11R6/include
+SYSCONF_CXXFLAGS_QT	= -I$(QTDIR)/include 
+SYSCONF_CXXFLAGS_OPENGL	= -I/usr/X11R6/include
+
+# Compiling YACC output
+SYSCONF_CXXFLAGS_YACC     = -Wno-unused -Wno-parentheses
+
+# Linking with support libraries
+SYSCONF_RPATH_X11	= 
+SYSCONF_RPATH_QT	= 
+SYSCONF_RPATH_OPENGL	= 
+
+# Linking with support libraries
+# X11
+SYSCONF_LFLAGS_X11	= -L/usr/X11R6/lib
+SYSCONF_LIBS_X11	= -lXext -lX11 
+# Qt, Qt+OpenGL
+SYSCONF_LFLAGS_QT	= -L$(QTDIR)/lib
+SYSCONF_LIBS_QT		= -lqt$(QT_THREAD_SUFFIX)
+SYSCONF_LIBS_QT_OPENGL	= 
+# OpenGL
+SYSCONF_LFLAGS_OPENGL	= -L/usr/X11R6/lib
+SYSCONF_LIBS_OPENGL	= -lGLU -lGL -lXmu
+# Yacc
+SYSCONF_LIBS_YACC	= 
+
+# Linking applications
+SYSCONF_LINK		= c++
+SYSCONF_LFLAGS		= 
+SYSCONF_LIBS		= 
+
+# Link flags for shared objects
+SYSCONF_LFLAGS_SHOBJ	= 
+
+# Flags for threading
+SYSCONF_CFLAGS_THREAD	= -pthread -D_THREAD_SAFE
+SYSCONF_CXXFLAGS_THREAD	= -pthread -D_THREAD_SAFE
+SYSCONF_LFLAGS_THREAD	= -pthread
+SYSCONF_LIBS_THREAD	= 
+
+# Meta-object compiler
+SYSCONF_MOC		= $(QTDIR)/bin/moc
+
+# UI compiler
+SYSCONF_UIC		= $(QTDIR)/bin/uic
+
+# Linking shared libraries
+#   - Build the $(TARGET) library, eg. lib$(TARGET).so.2.2.2
+#   - Place target in $(DESTDIR) - which has a trailing /
+#   - Usually needs to incorporate $(VER_MAJ), $(VER_MIN) and $(VER_PATCH)
+#
+SYSCONF_LINK_SHLIB	= c++
+SYSCONF_LINK_TARGET_SHARED	= lib$(TARGET).$(VER_MAJ).$(VER_MIN).$(VER_PATCH).dylib
+SYSCONF_LINK_LIB_SHARED	=  $(SYSCONF_LINK_SHLIB) -dynamiclib -undefined suppress \
+				     -compatibility_version $(VER_MAJ).$(VER_MIN) \
+				     -current_version $(VER_MAJ).$(VER_MIN).$(VER_PATCH) \
+				     -install_name $(FINAL_LIB_DIR)lib$(TARGET).$(VER_MAJ).dylib \
+				     $(LFLAGS) -o $(SYSCONF_LINK_TARGET_SHARED) \
+				     $(OBJECTS) $(OBJMOC) $(LIBS) && \
+				 mv $(SYSCONF_LINK_TARGET_SHARED) $(DESTDIR); \
+				 cd $(DESTDIR) && \
+				 rm -f lib$(TARGET).dylib lib$(TARGET).$(VER_MAJ).dylib; \
+				 ln -s $(SYSCONF_LINK_TARGET_SHARED) lib$(TARGET).dylib; \
+				 ln -s $(SYSCONF_LINK_TARGET_SHARED) lib$(TARGET).$(VER_MAJ).dylib
+
+# Linking static libraries
+#   - Build the $(TARGET) library, eg. lib$(TARGET).a
+#   - Place target in $(DESTDIR) - which has a trailing /
+#
+SYSCONF_AR		= ar cq
+SYSCONF_RANLIB		= ranlib 
+SYSCONF_LINK_TARGET_STATIC = lib$(TARGET).a
+SYSCONF_LINK_LIB_STATIC	= rm -f $(DESTDIR)$(SYSCONF_LINK_TARGET_STATIC) ; \
+				 $(SYSCONF_AR) $(DESTDIR)$(SYSCONF_LINK_TARGET_STATIC) $(OBJECTS) $(OBJMOC) ; \
+				 $(SYSCONF_RANLIB) $(DESTDIR)$(SYSCONF_LINK_TARGET_STATIC) \
+# Compiling application source
+SYSCONF_CXXFLAGS	= -pipe -fno-exceptions -O2 
+SYSCONF_CFLAGS		= -pipe -fno-exceptions -O2 
+# Default link type (static linking is still be used where required)
+SYSCONF_LINK_LIB	= $(SYSCONF_LINK_LIB_SHARED)
+SYSCONF_LINK_TARGET	= $(SYSCONF_LINK_TARGET_SHARED)
+# Compiling library source
+SYSCONF_CXXFLAGS_LIB	= -fno-common
+SYSCONF_CFLAGS_LIB	= -fno-common
+# Compiling shared-object source
+SYSCONF_CXXFLAGS_SHOBJ	= -dynamic
+SYSCONF_CFLAGS_SHOBJ	= -dynamic
+# Linking Qt
+SYSCONF_LIBS_QTLIB	= $(SYSCONF_CXXFLAGS_X11) $(QT_LIBS_MT) $(QT_LIBS_OPT)
+# Linking Qt applications
+SYSCONF_LIBS_QTAPP	= $(SYSCONF_LFLAGS_X11) $(SYSCONF_CXXFLAGS_X11) $(QT_LIBS_MT) $(QT_LIBS_OPT) $(SYSCONF_LIBS_X11)
diff -ruN qt-2.3.0/configs/darwin-c++-static qt-2.3.0-patched/configs/darwin-c++-static
--- qt-2.3.0/configs/darwin-c++-static	Thu Jan  1 01:00:00 1970
+++ qt-2.3.0-patched/configs/darwin-c++-static	Fri Mar 16 10:46:40 2001
@@ -0,0 +1,95 @@
+# Compiling
+INTERFACE_DECL_PATH 	= .
+SYSCONF_CXX		= c++
+SYSCONF_CC		= cc
+DASHCROSS		= 
+
+# Compiling with support libraries
+SYSCONF_CXXFLAGS_X11	= -I/usr/X11R6/include
+SYSCONF_CXXFLAGS_QT	= -I$(QTDIR)/include 
+SYSCONF_CXXFLAGS_OPENGL	= -I/usr/X11R6/include
+
+# Compiling YACC output
+SYSCONF_CXXFLAGS_YACC     = -Wno-unused -Wno-parentheses
+
+# Linking with support libraries
+SYSCONF_RPATH_X11	= 
+SYSCONF_RPATH_QT	= 
+SYSCONF_RPATH_OPENGL	= 
+
+# Linking with support libraries
+# X11
+SYSCONF_LFLAGS_X11	= -L/usr/X11R6/lib
+SYSCONF_LIBS_X11	= -lXext -lX11 
+# Qt, Qt+OpenGL
+SYSCONF_LFLAGS_QT	= -L$(QTDIR)/lib
+SYSCONF_LIBS_QT		= -lqt$(QT_THREAD_SUFFIX)
+SYSCONF_LIBS_QT_OPENGL	= 
+# OpenGL
+SYSCONF_LFLAGS_OPENGL	= -L/usr/X11R6/lib
+SYSCONF_LIBS_OPENGL	= -lGLU -lGL -lXmu
+# Yacc
+SYSCONF_LIBS_YACC	= 
+
+# Linking applications
+SYSCONF_LINK		= c++
+SYSCONF_LFLAGS		= 
+SYSCONF_LIBS		= 
+
+# Link flags for shared objects
+SYSCONF_LFLAGS_SHOBJ	= -shared
+
+# Flags for threading
+SYSCONF_CFLAGS_THREAD	= -pthread -D_THREAD_SAFE
+SYSCONF_CXXFLAGS_THREAD	= -pthread -D_THREAD_SAFE
+SYSCONF_LFLAGS_THREAD	= -pthread
+SYSCONF_LIBS_THREAD	= 
+
+# Meta-object compiler
+SYSCONF_MOC		= $(QTDIR)/bin/moc
+
+# UI compiler
+SYSCONF_UIC		= $(QTDIR)/bin/uic
+
+# Linking shared libraries
+#   - Build the $(TARGET) library, eg. lib$(TARGET).so.2.2.2
+#   - Place target in $(DESTDIR) - which has a trailing /
+#   - Usually needs to incorporate $(VER_MAJ), $(VER_MIN) and $(VER_PATCH)
+#
+SYSCONF_LINK_SHLIB	= c++
+SYSCONF_LINK_TARGET_SHARED	= lib$(TARGET).dylib.$(VER_MAJ).$(VER_MIN).$(VER_PATCH)
+SYSCONF_LINK_LIB_SHARED	=  $(SYSCONF_LINK_SHLIB) -shared  \
+				     $(LFLAGS) -o $(SYSCONF_LINK_TARGET_SHARED) \
+				     $(OBJECTS) $(OBJMOC) $(LIBS) && \
+				 mv $(SYSCONF_LINK_TARGET_SHARED) $(DESTDIR); \
+				 cd $(DESTDIR) && \
+				 rm -f lib$(TARGET).dylib lib$(TARGET).dylib.$(VER_MAJ) lib$(TARGET).dylib.$(VER_MAJ).$(VER_MIN); \
+				 ln -s $(SYSCONF_LINK_TARGET_SHARED) lib$(TARGET).dylib; \
+				 ln -s $(SYSCONF_LINK_TARGET_SHARED) lib$(TARGET).dylib.$(VER_MAJ); \
+				 ln -s $(SYSCONF_LINK_TARGET_SHARED) lib$(TARGET).dylib.$(VER_MAJ).$(VER_MIN)
+
+# Linking static libraries
+#   - Build the $(TARGET) library, eg. lib$(TARGET).a
+#   - Place target in $(DESTDIR) - which has a trailing /
+#
+SYSCONF_AR		= ar cq
+SYSCONF_RANLIB		= ranlib 
+SYSCONF_LINK_TARGET_STATIC = lib$(TARGET).a
+SYSCONF_LINK_LIB_STATIC	= rm -f $(DESTDIR)$(SYSCONF_LINK_TARGET_STATIC) ; \
+				 $(SYSCONF_AR) $(DESTDIR)$(SYSCONF_LINK_TARGET_STATIC) $(OBJECTS) $(OBJMOC) ; \
+				 $(SYSCONF_RANLIB) $(DESTDIR)$(SYSCONF_LINK_TARGET_STATIC) \
+# Compiling application source
+SYSCONF_CXXFLAGS	= -pipe -fno-exceptions -O2 
+SYSCONF_CFLAGS		= -pipe -fno-exceptions -O2 
+SYSCONF_LINK_LIB	= $(SYSCONF_LINK_LIB_STATIC)
+SYSCONF_LINK_TARGET	= $(SYSCONF_LINK_TARGET_STATIC)
+# Compiling library source
+SYSCONF_CXXFLAGS_LIB	=
+SYSCONF_CFLAGS_LIB	=
+# Compiling shared-object source
+SYSCONF_CXXFLAGS_SHOBJ	= -fPIC
+SYSCONF_CFLAGS_SHOBJ	= -fPIC
+# Linking Qt
+SYSCONF_LIBS_QTLIB	= $(SYSCONF_CXXFLAGS_X11) $(QT_LIBS_MT) $(QT_LIBS_OPT)
+# Linking Qt applications
+SYSCONF_LIBS_QTAPP	= $(SYSCONF_LFLAGS_X11) $(SYSCONF_CXXFLAGS_X11) $(QT_LIBS_MT) $(QT_LIBS_OPT) $(SYSCONF_LIBS_X11)
diff -ruN qt-2.3.0/configure qt-2.3.0-patched/configure
--- qt-2.3.0/configure	Mon Mar  5 19:12:49 2001
+++ qt-2.3.0-patched/configure	Fri Mar 16 10:46:40 2001
@@ -6,7 +6,7 @@
 #
 
 # To remove the need for the first test, do this:
-# QTDIR=`pwd` export QTDIR
+QTDIR=`pwd` export QTDIR
 
 relpath=`dirname $0`
 relpath=`(cd $relpath; pwd)`
@@ -1209,7 +1209,7 @@
 fi
 
 [ "x$SM" = "xyes" ] && QT_LIBS="${QT_LIBS} -lSM -lICE"
-[ "x$XFT" = "xyes" ] && QT_LIBS="${QT_LIBS} -lXft"
+[ "x$XFT" = "xyes" ] && QT_LIBS="${QT_LIBS} -lXft -lXrender"
 if [ -z "$EMB" -o "$SHARED" = no ]
 then
     [ "x$LIBPNG" = "xyes" ] && QT_MODOBJ=$QT_MODOBJ' $(PNG_OBJECTS)'
diff -ruN qt-2.3.0/include/qglobal.h qt-2.3.0-patched/include/qglobal.h
--- qt-2.3.0/include/qglobal.h	Mon Mar  5 19:09:36 2001
+++ qt-2.3.0-patched/include/qglobal.h	Fri Mar 16 10:46:40 2001
@@ -73,6 +73,8 @@
 
 #if defined(macintosh)
 #define _OS_MAC_
+#elif defined(__APPLE__) && defined(__ppc__)
+#define _OS_DARWIN_
 #elif defined(MSDOS) || defined(_MSDOS) || defined(__MSDOS__)
 #define _OS_MSDOS_
 #elif defined(OS2) || defined(_OS2) || defined(__OS2__)
diff -ruN qt-2.3.0/src/network/qsocketdevice_unix.cpp qt-2.3.0-patched/src/network/qsocketdevice_unix.cpp
--- qt-2.3.0/src/network/qsocketdevice_unix.cpp	Mon Mar  5 19:09:36 2001
+++ qt-2.3.0-patched/src/network/qsocketdevice_unix.cpp	Fri Mar 16 10:46:40 2001
@@ -146,8 +146,12 @@
 // new linux is Single Unix 1998, not old linux
 #define SOCKLEN_T socklen_t
 #elif defined(BSD4_4)
-// freebsd
+// freebsd or Darwin
+#if defined(_OS_DARWIN_)
+#define SOCKLEN_T int
+#else
 #define SOCKLEN_T socklen_t
+#endif
 #elif defined(_OS_UNIXWARE7_)
 #define SOCKLEN_T size_t
 #elif defined(_OS_AIX_)
diff -ruN qt-2.3.0/src/tools/qglobal.h qt-2.3.0-patched/src/tools/qglobal.h
--- qt-2.3.0/src/tools/qglobal.h	Mon Mar  5 19:09:36 2001
+++ qt-2.3.0-patched/src/tools/qglobal.h	Fri Mar 16 10:46:40 2001
@@ -73,6 +73,8 @@
 
 #if defined(macintosh)
 #define _OS_MAC_
+#elif defined(__APPLE__) && defined(__ppc__)
+#define _OS_DARWIN_
 #elif defined(MSDOS) || defined(_MSDOS) || defined(__MSDOS__)
 #define _OS_MSDOS_
 #elif defined(OS2) || defined(_OS2) || defined(__OS2__)
