--- screen-3.9.8/Makefile.in	Mon Feb 26 16:07:53 2001
+++ screen-3.9.8/Makefile.in	Mon Feb 26 16:07:57 2001
@@ -71,7 +71,7 @@
 	-if [ -f $(bindir)/$(SCREEN) ] && [ ! -f $(bindir)/$(SCREEN).old ]; \
 		then mv $(bindir)/$(SCREEN) $(bindir)/$(SCREEN).old; fi
 	$(INSTALL_PROGRAM) screen $(bindir)/$(SCREEN)
-	-chown root $(bindir)/$(SCREEN) && chmod 4755 $(bindir)/$(SCREEN)
+#       -chown root $(bindir)/$(SCREEN) && chmod 4755 $(bindir)/$(SCREEN)
 # This doesn't work if $(bindir)/screen is a symlink
 	-if [ -f $(bindir)/screen ] && [ ! -f $(bindir)/screen.old ]; then mv $(bindir)/screen $(bindir)/screen.old; fi
 	rm -f $(bindir)/screen
