--- Makefile1	Thu Feb 15 19:29:28 2001
+++ Makefile1	Thu Feb 15 19:29:29 2001
@@ -80,13 +80,6 @@
                 $(MAKE) $@) || exit 1; \
 	done
 	$(srcdir)/selfautofix "$(bindir)" "$(texmf)/web2c"
-	cd $(bindir); for i in $(ALLPROGS); do test -f $$i && { echo strip $$i; strip $$i || true; }; done; true
-	test -f $(scriptdir)/mktexlsr && \
-	  TEXMFMAIN=$(texmf) PATH=$(scriptdir):$(bindir):$(PATH) \
-	    $(scriptdir)/mktexlsr
-	test -f $(scriptdir)/texconfig && \
-	  TEXMFMAIN=$(texmf) PATH=$(scriptdir):$(bindir):$(PATH) \
-	    $(scriptdir)/texconfig init
 
 install-exec:
 	for dir in $(ESUBDIRS); do \
--- Makefile2	Thu Feb 15 19:29:28 2001
+++ Makefile2	Thu Feb 15 19:29:31 2001
@@ -74,19 +74,10 @@
 
 default: all
 install: all 
-	for dir in $(ESUBDIRS); do \
-	   echo making $@ in $$dir; \
-	     (cd $$dir && PATH=$(scriptdir):$(bindir):$(PATH) \
-                $(MAKE) $@) || exit 1; \
-	done
-	$(srcdir)/selfautofix "$(bindir)" "$(texmf)/web2c"
 	cd $(bindir); for i in $(ALLPROGS); do test -f $$i && { echo strip $$i; strip $$i || true; }; done; true
 	test -f $(scriptdir)/mktexlsr && \
 	  TEXMFMAIN=$(texmf) PATH=$(scriptdir):$(bindir):$(PATH) \
 	    $(scriptdir)/mktexlsr
-	test -f $(scriptdir)/texconfig && \
-	  TEXMFMAIN=$(texmf) PATH=$(scriptdir):$(bindir):$(PATH) \
-	    $(scriptdir)/texconfig init
 
 install-exec:
 	for dir in $(ESUBDIRS); do \
