diff -ru zlib-1.1.3/configure zlib-1.1.3-patched/configure
--- zlib-1.1.3/configure	Wed Jul  8 20:19:35 1998
+++ zlib-1.1.3-patched/configure	Mon Jan 15 10:34:26 2001
@@ -116,6 +116,11 @@
 	     SFLAGS=${CFLAGS-"-Kconform_pic -O"}
 	     CFLAGS=${CFLAGS-"-O"}
 	     LDSHARED=${LDSHARED-"cc -G"};;
+  Darwin*)   SFLAGS=${CFLAGS-"-O2 -fno-common"}
+             CFLAGS=${CFLAGS-"-O2"}
+	     LDSHARED=${LDSHARED-"cc -dynamiclib -undefined suppress -install_name $libdir/libz.dylib"}
+	     shared_ext='.dylib'
+	     SHAREDLIB='libz.dylib';;
   # send working options for other systems to support@gzip.org
   *)         SFLAGS=${CFLAGS-"-O"}
 	     CFLAGS=${CFLAGS-"-O"}
