diff -ruN gettext-0.22.5-orig/gettext-tools/tests/lang-bash gettext-0.22.5/gettext-tools/tests/lang-bash
--- gettext-0.22.5-orig/gettext-tools/tests/lang-bash	2019-05-11 06:29:32.000000000 -0500
+++ gettext-0.22.5/gettext-tools/tests/lang-bash	2024-12-02 03:50:42.000000000 -0600
@@ -14,6 +14,8 @@
 # because in this case the gettext-runtime/src/gettext program does not do
 # any message catalog lookups.
 
+echo "Skipping test: Darwin's bash lacks iconv support"; rm -fr $tmpfiles; exit 77;
+
 cat <<\EOF > prog.bash
 #! /bin/bash
 
diff -ruN gettext-0.22.5-orig/gettext-tools/tests/lang-vala gettext-0.22.5/gettext-tools/tests/lang-vala
--- gettext-0.22.5-orig/gettext-tools/tests/lang-vala	2022-10-09 02:30:42.000000000 -0500
+++ gettext-0.22.5/gettext-tools/tests/lang-vala	2024-12-02 03:50:42.000000000 -0600
@@ -40,7 +40,7 @@
 EOF
 
 : ${VALAC=valac}
-${VALAC} --Xcc=-DGETTEXT_PACKAGE=\"prog\" prog.vala 2>prog.err \
+${VALAC} --Xcc="-I@FINKPREFIX@/include" --Xcc=-DGETTEXT_PACKAGE=\"prog\" prog.vala 2>prog.err \
   || { cat prog.err 1>&2; Exit 1; }
 
 : ${XGETTEXT=xgettext}
