diff -uNr gzip-1.6.orig/lib/vasnprintf.c gzip-1.6/lib/vasnprintf.c
--- gzip-1.6.orig/lib/vasnprintf.c	2013-06-09 16:40:51.000000000 -0400
+++ gzip-1.6/lib/vasnprintf.c	2017-08-13 09:51:43.000000000 -0400
@@ -4870,7 +4870,7 @@
 #endif
                   *fbp = dp->conversion;
 #if USE_SNPRINTF
-# if !(((__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 3)) && !defined __UCLIBC__) || ((defined _WIN32 || defined __WIN32__) && ! defined __CYGWIN__))
+# if !defined(__APPLE__) && !(((__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 3)) && !defined __UCLIBC__) || ((defined _WIN32 || defined __WIN32__) && ! defined __CYGWIN__))
                 fbp[1] = '%';
                 fbp[2] = 'n';
                 fbp[3] = '\0';
