diff -ruN gettext-0.22.5-orig/gettext-runtime/libasprintf/Makefile.in gettext-0.22.5/gettext-runtime/libasprintf/Makefile.in
--- gettext-0.22.5-orig/gettext-runtime/libasprintf/Makefile.in	2024-02-21 10:35:57.000000000 -0600
+++ gettext-0.22.5/gettext-runtime/libasprintf/Makefile.in	2024-12-02 03:18:37.000000000 -0600
@@ -1333,7 +1333,7 @@
 # How to build libasprintf.
 # With libtool 1.5.14, on some platforms, like BeOS, "libtool --tag=CXX" fails
 # to create a shared library, however "libtool --tag=CC" succeeds.
-libasprintf_la_LDFLAGS = @LTNOUNDEF@ -export-symbols-regex $(LIBASPRINTF_EXPORTED_SYMBOLS_REGEX)
+libasprintf_la_LDFLAGS = @LTNOUNDEF@ -static -export-symbols-regex $(LIBASPRINTF_EXPORTED_SYMBOLS_REGEX)
 
 # Documentation.
 
diff -ruN gettext-0.22.5-orig/gettext-tools/src/Makefile.in gettext-0.22.5/gettext-tools/src/Makefile.in
--- gettext-0.22.5-orig/gettext-tools/src/Makefile.in	2024-02-21 10:36:32.000000000 -0600
+++ gettext-0.22.5/gettext-tools/src/Makefile.in	2024-12-02 03:18:37.000000000 -0600
@@ -4940,10 +4940,10 @@
 	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(urlget_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o urlget-urlget.obj `if test -f 'urlget.c'; then $(CYGPATH_W) 'urlget.c'; else $(CYGPATH_W) '$(srcdir)/urlget.c'; fi`
 
 xgettext-xgettext.o: xgettext.c
-	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(xgettext_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o xgettext-xgettext.o `test -f 'xgettext.c' || echo '$(srcdir)/'`xgettext.c
+	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(xgettext_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -O0 -c -o xgettext-xgettext.o `test -f 'xgettext.c' || echo '$(srcdir)/'`xgettext.c
 
 xgettext-xgettext.obj: xgettext.c
-	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(xgettext_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o xgettext-xgettext.obj `if test -f 'xgettext.c'; then $(CYGPATH_W) 'xgettext.c'; else $(CYGPATH_W) '$(srcdir)/xgettext.c'; fi`
+	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(xgettext_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -O0 -c -o xgettext-xgettext.obj `if test -f 'xgettext.c'; then $(CYGPATH_W) 'xgettext.c'; else $(CYGPATH_W) '$(srcdir)/xgettext.c'; fi`
 
 xgettext-xg-pos.o: xg-pos.c
 	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(xgettext_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o xgettext-xg-pos.o `test -f 'xg-pos.c' || echo '$(srcdir)/'`xg-pos.c
