diff -Nurd libintl-perl-1.32.orig/Makefile.PL libintl-perl-1.32/Makefile.PL
--- libintl-perl-1.32.orig/Makefile.PL	2018-10-08 04:37:35.000000000 -0400
+++ libintl-perl-1.32/Makefile.PL	2022-01-21 14:15:09.000000000 -0500
@@ -92,7 +92,10 @@
 }
 EOF
 
-my $build_gettext_xs = ac_try_link $c, "Checking whether we can compile the XS version";
+my $want_xs = -1; # edit this line: -1=no, 0=if-found, 1=require-yes
+my $build_gettext_xs = 0;
+if ($want_xs != -1) {
+$build_gettext_xs = ac_try_link $c, "Checking whether we can compile the XS version";
 unless ($build_gettext_xs) {
     $build_gettext_xs = ac_try_link $c, "  Trying again with -lintl", "-lintl";
 }
@@ -102,12 +105,24 @@
 unless ($build_gettext_xs) {
     $build_gettext_xs = ac_try_link $c, "  Trying again with -liconv", "-liconv";
 }
+}
 
 unless ($build_gettext_xs) {
+if ($want_xs == 0) {
 	print STDERR <<EOF;
 => Cannot build the XS version of libintl-perl (see 'config.log' for
 => details).  This is harmless!
 EOF
+} elsif ($want_xs == 1) {
+	die <<EOF;
+=> Cannot build the XS version of libintl-perl (see 'config.log' for
+=> details).  But build-time option requires it!
+EOF
+} else { # $want_xs == -1
+	print STDERR <<EOF;
+=> Build-time option disabled the XS version of libintl-perl.
+EOF
+}
 } else {
 	print STDERR <<EOF;
 => Building the XS version of libintl-perl.
@@ -124,6 +139,7 @@
     close HANDLE or die "cannot close file 'gettest.c': $!\n";
 
 	my @cmd = $Config{cc};
+	push @cmd, ('-IFINKPATH/include', '-LFINKPATH/lib');
     push @cmd, $Config{ccflags};
     push @cmd, $Config{ldflags};
     push @cmd, $Config{perllibs};
diff -Nurd libintl-perl-1.32.orig/gettext_xs/Makefile.PL libintl-perl-1.32/gettext_xs/Makefile.PL
--- libintl-perl-1.32.orig/gettext_xs/Makefile.PL	2018-10-08 04:37:35.000000000 -0400
+++ libintl-perl-1.32/gettext_xs/Makefile.PL	2022-01-21 13:47:12.000000000 -0500
@@ -37,7 +37,8 @@
     ),
     PREREQ_PM => { File::Spec => 0 },
 	PL_FILES        => {},
-	LIBS => ['-lintl -liconv'],
+	LIBS => ['-LFINKPATH/lib -lintl'],
+	INC => '-IFINKPATH/include',
 );
 
 __END__
