diff -Nurd -x'*~' libintl-perl-1.32.orig/Makefile.PL libintl-perl-1.32/Makefile.PL
--- libintl-perl-1.32.orig/Makefile.PL	2018-10-08 04:37:35.000000000 -0400
+++ libintl-perl-1.32/Makefile.PL	2020-12-11 01:21:33.000000000 -0500
@@ -104,9 +104,9 @@
 }
 
 unless ($build_gettext_xs) {
-	print STDERR <<EOF;
+	die <<EOF;
 => Cannot build the XS version of libintl-perl (see 'config.log' for
-=> details).  This is harmless!
+=> details).
 EOF
 } else {
 	print STDERR <<EOF;
@@ -124,6 +124,7 @@
     close HANDLE or die "cannot close file 'gettest.c': $!\n";
 
 	my @cmd = $Config{cc};
+	push @cmd, ('-IFINKPATH/include', '-LFINKPATH/lib');
     push @cmd, $Config{ccflags};
     push @cmd, $Config{ldflags};
     push @cmd, $Config{perllibs};
diff -Nurd -x'*~' libintl-perl-1.32.orig/gettext_xs/Makefile.PL libintl-perl-1.32/gettext_xs/Makefile.PL
--- libintl-perl-1.32.orig/gettext_xs/Makefile.PL	2018-10-08 04:37:35.000000000 -0400
+++ libintl-perl-1.32/gettext_xs/Makefile.PL	2020-12-11 01:25:48.000000000 -0500
@@ -37,7 +37,8 @@
     ),
     PREREQ_PM => { File::Spec => 0 },
 	PL_FILES        => {},
-	LIBS => ['-lintl -liconv'],
+	LIBS => ['-LFINKPATH/lib -lintl'],
+	INC => '-IFINKPATH/include',
 );
 
 __END__
