diff -ruN gettext-0.14.5/gettext-tools/lib/execute.c gettext-0.14.5-patched/gettext-tools/lib/execute.c
--- gettext-0.14.5/gettext-tools/lib/execute.c	2005-05-20 13:19:48.000000000 -0700
+++ gettext-0.14.5-patched/gettext-tools/lib/execute.c	2006-08-14 05:21:26.000000000 -0700
@@ -62,7 +62,12 @@
 #endif
 
 #ifndef HAVE_ENVIRON_DECL
-extern char **environ;
+# ifdef __APPLE__
+#  include <crt_externs.h>
+#  define environ (*_NSGetEnviron())
+# else
+   extern char **environ;
+# endif
 #endif
 
 #ifndef STDIN_FILENO
diff -ruN gettext-0.14.5/gettext-tools/lib/pipe.c gettext-0.14.5-patched/gettext-tools/lib/pipe.c
--- gettext-0.14.5/gettext-tools/lib/pipe.c	2005-05-20 13:25:23.000000000 -0700
+++ gettext-0.14.5-patched/gettext-tools/lib/pipe.c	2006-08-14 05:22:02.000000000 -0700
@@ -61,7 +61,12 @@
 #endif
 
 #ifndef HAVE_ENVIRON_DECL
-extern char **environ;
+# ifdef __APPLE__
+#  include <crt_externs.h>
+#  define environ (*_NSGetEnviron())
+# else
+   extern char **environ;
+# endif
 #endif
 
 #ifndef STDIN_FILENO
