diff -ruN apt/apt-pkg/deb/debindexfile.cc apt-patched/apt-pkg/deb/debindexfile.cc
--- apt/apt-pkg/deb/debindexfile.cc	Tue Feb 27 05:14:22 2001
+++ apt-patched/apt-pkg/deb/debindexfile.cc	Wed Aug 22 18:26:23 2001
@@ -496,3 +496,12 @@
 }
 
 									/*}}}*/
+
+void init_deb2()
+{
+  (void)_apt_DebType;
+  (void)_apt_DebSrcType;
+  (void)_apt_Src;
+  (void)_apt_Pkg;
+  (void)_apt_Status;
+}
diff -ruN apt/apt-pkg/deb/debsystem.cc apt-patched/apt-pkg/deb/debsystem.cc
--- apt/apt-pkg/deb/debsystem.cc	Tue Feb 20 08:03:17 2001
+++ apt-patched/apt-pkg/deb/debsystem.cc	Wed Aug 22 18:26:23 2001
@@ -30,6 +30,15 @@
 
 debSystem debSys;
 
+extern void init_deb2();
+extern void init_deb3();
+void init_deb()
+{
+  (void)debSys;
+  init_deb2();
+  init_deb3();
+}
+
 // System::debSystem - Constructor					/*{{{*/
 // ---------------------------------------------------------------------
 /* */
@@ -153,8 +162,8 @@
       which is yet to be determined. The functions in pkgcachegen should
       be the only users of these */
    Cnf.CndSet("Dir::State::userstatus","status.user"); // Defunct
-   Cnf.CndSet("Dir::State::status","/var/lib/dpkg/status");
-   Cnf.CndSet("Dir::Bin::dpkg","/usr/bin/dpkg");
+   Cnf.CndSet("Dir::State::status","@PREFIX@/var/lib/dpkg/status");
+   Cnf.CndSet("Dir::Bin::dpkg","@PREFIX@/bin/dpkg");
    
    return true;
 }
@@ -177,9 +186,9 @@
 signed debSystem::Score(Configuration const &Cnf)
 {
    signed Score = 0;
-   if (FileExists(Cnf.FindFile("Dir::State::status","/var/lib/dpkg/status")) == true)
+   if (FileExists(Cnf.FindFile("Dir::State::status","@PREFIX@/var/lib/dpkg/status")) == true)
        Score += 10;
-   if (FileExists(Cnf.FindFile("Dir::Bin::dpkg","/usr/bin/dpkg")) == true)
+   if (FileExists(Cnf.FindFile("Dir::Bin::dpkg","@PREFIX@/bin/dpkg")) == true)
       Score += 10;
    if (FileExists("/etc/debian_version") == true)
       Score += 10;
diff -ruN apt/apt-pkg/deb/debversion.cc apt-patched/apt-pkg/deb/debversion.cc
--- apt/apt-pkg/deb/debversion.cc	Tue Feb 20 08:03:17 2001
+++ apt-patched/apt-pkg/deb/debversion.cc	Wed Aug 22 18:26:23 2001
@@ -23,6 +23,11 @@
 
 debVersioningSystem debVS;
 
+void init_deb3()
+{
+  (void)debVS;
+}
+
 // debVS::debVersioningSystem - Constructor				/*{{{*/
 // ---------------------------------------------------------------------
 /* */
diff -ruN apt/apt-pkg/init.cc apt-patched/apt-pkg/init.cc
--- apt/apt-pkg/init.cc	Tue Feb 20 08:03:17 2001
+++ apt-patched/apt-pkg/init.cc	Wed Aug 22 18:26:23 2001
@@ -15,6 +15,8 @@
 #include <apti18n.h>
 #include <config.h>
 #include <sys/stat.h>
+
+extern void init_deb();
 									/*}}}*/
 
 #define Stringfy_(x) # x
@@ -39,7 +41,7 @@
       Cnf.Set("APT::Architecture",COMMON_CPU);
    else
       Cnf.Set("APT::Architecture",COMMON_OS "-" COMMON_CPU);
-   Cnf.Set("Dir","/");
+   Cnf.Set("Dir","@PREFIX@/");
    
    // State   
    Cnf.Set("Dir::State","var/lib/apt/");
@@ -66,7 +68,7 @@
    Cnf.Set("Dir::Etc::main","apt.conf");
    Cnf.Set("Dir::Etc::parts","apt.conf.d");
    Cnf.Set("Dir::Etc::preferences","preferences");
-   Cnf.Set("Dir::Bin::methods","/usr/lib/apt/methods");
+   Cnf.Set("Dir::Bin::methods","@PREFIX@/lib/apt/methods");
 
    bool Res = true;
    
@@ -99,6 +101,8 @@
 /* */
 bool pkgInitSystem(Configuration &Cnf,pkgSystem *&Sys)
 {
+   init_deb();
+
    Sys = 0;
    string Label = Cnf.Find("Apt::System","");
    if (Label.empty() == false)
diff -ruN apt/buildlib/environment.mak.in apt-patched/buildlib/environment.mak.in
--- apt/buildlib/environment.mak.in	Fri Feb 23 08:25:26 2001
+++ apt-patched/buildlib/environment.mak.in	Wed Aug 22 18:26:23 2001
@@ -11,7 +11,7 @@
 LIBSTDCPP_VER = @LIBSTDCPP_VER@
 
 # Linker stuff
-PICFLAGS+= -fPIC -DPIC
+PICFLAGS+= -fno-common -DPIC
 LFLAGS+= @LDFLAGS@
 LEFLAGS+= 
 SOCKETLIBS:= @SOCKETLIBS@
@@ -46,11 +46,13 @@
 
 # Shared library things
 HOST_OS = @host_os@
-ifneq ($(words $(filter linux-gnu gnu%,$(HOST_OS))),0)
-   SONAME_MAGIC=-Wl,-soname -Wl,
-   LFLAGS_SO=
-else
-   # Do not know how to create shared libraries here.
-   ONLYSTATICLIBS = yes
-endif
+#ifneq ($(words $(filter linux-gnu gnu%,$(HOST_OS))),0)
+#   SONAME_MAGIC=-Wl,-soname -Wl,
+#   LFLAGS_SO=
+#else
+#   # Do not know how to create shared libraries here.
+#   ONLYSTATICLIBS = yes
+#endif
+SONAME_MAGIC=-install_name @PREFIX@/lib/
+LFLAGS_SO=-dynamiclib -flat_namespace -undefined suppress
 	
diff -ruN apt/buildlib/library.mak apt-patched/buildlib/library.mak
--- apt/buildlib/library.mak	Sun Feb 25 02:43:14 2001
+++ apt-patched/buildlib/library.mak	Wed Aug 22 18:26:23 2001
@@ -15,17 +15,17 @@
 # See defaults.mak for information about LOCAL
 
 # Some local definitions
-LOCAL := lib$(LIBRARY)$(LIBEXT).so.$(MAJOR).$(MINOR)
+LOCAL := lib$(LIBRARY)$(LIBEXT).$(MAJOR).$(MINOR).dylib
 $(LOCAL)-OBJS := $(addprefix $(OBJ)/,$(addsuffix .opic,$(notdir $(basename $(SOURCE)))))
 $(LOCAL)-DEP := $(addprefix $(DEP)/,$(addsuffix .opic.d,$(notdir $(basename $(SOURCE)))))
 $(LOCAL)-HEADERS := $(addprefix $(INCLUDE)/,$(HEADERS))
-$(LOCAL)-SONAME := lib$(LIBRARY)$(LIBEXT).so.$(MAJOR)
+$(LOCAL)-SONAME := lib$(LIBRARY)$(LIBEXT).$(MAJOR).dylib
 $(LOCAL)-SLIBS := $(SLIBS)
 $(LOCAL)-LIBRARY := $(LIBRARY)
 
 # Install the command hooks
 headers: $($(LOCAL)-HEADERS)
-library: $(LIB)/lib$(LIBRARY).so $(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR)
+library: $(LIB)/lib$(LIBRARY).dylib $(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).dylib
 clean: clean/$(LOCAL)
 veryclean: veryclean/$(LOCAL)
 
@@ -37,21 +37,23 @@
 clean/$(LOCAL):
 	-rm -f $($(@F)-OBJS) $($(@F)-DEP)
 veryclean/$(LOCAL): clean/$(LOCAL)
-	-rm -f $($(@F)-HEADERS) $(LIB)/lib$($(@F)-LIBRARY)*.so*
+	-rm -f $($(@F)-HEADERS) $(LIB)/lib$($(@F)-LIBRARY)*.dylib
 
 # Build rules for the two symlinks
-.PHONY: $(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR) $(LIB)/lib$(LIBRARY).so
-$(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR): $(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR).$(MINOR)
+.PHONY: $(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).dylib $(LIB)/lib$(LIBRARY).dylib
+$(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).dylib: $(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).$(MINOR).dylib
 	ln -sf $(<F) $@
-$(LIB)/lib$(LIBRARY).so: $(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR).$(MINOR)
+$(LIB)/lib$(LIBRARY).dylib: $(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).$(MINOR).dylib
 	ln -sf $(<F) $@
 	
 # The binary build rule
-$(LIB)/lib$(LIBRARY)$(LIBEXT).so.$(MAJOR).$(MINOR): $($(LOCAL)-HEADERS) $($(LOCAL)-OBJS)
-	-rm -f $(LIB)/lib$($(@F)-LIBRARY)*.so* 2> /dev/null
+$(LIB)/lib$(LIBRARY)$(LIBEXT).$(MAJOR).$(MINOR).dylib: $($(LOCAL)-HEADERS) $($(LOCAL)-OBJS)
+	-rm -f $(LIB)/lib$($(@F)-LIBRARY)*.dylib 2> /dev/null
 	echo Building shared library $@
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(PICFLAGS) $(LFLAGS) $(LFLAGS_SO)\
-	   -o $@ $(SONAME_MAGIC)$($(@F)-SONAME) -shared \
+	   -o $@ $(SONAME_MAGIC)$($(@F)-SONAME) \
+	   -compatibility_version $(MAJOR).$(MINOR) \
+	   -current_version $(MAJOR).$(MINOR) \
 	   $(filter %.opic,$^) \
 	   $($(@F)-SLIBS) 
 
diff -ruN apt/buildlib/ostable apt-patched/buildlib/ostable
--- apt/buildlib/ostable	Tue Feb 20 08:03:17 2001
+++ apt-patched/buildlib/ostable	Wed Aug 22 18:26:23 2001
@@ -14,6 +14,7 @@
 hp-hpux[^-]*	    hp-ux
 sun-solaris[^-]*    solaris
 [^-]*-openbsd[^-]*  openbsd
+[^-]*-darwin[^-]*   darwin
 
 # Catch all
 .*	unknown
diff -ruN apt/cmdline/apt-cache.cc apt-patched/cmdline/apt-cache.cc
--- apt/cmdline/apt-cache.cc	Thu Mar  8 03:20:43 2001
+++ apt-patched/cmdline/apt-cache.cc	Wed Aug 22 18:26:23 2001
@@ -371,9 +371,11 @@
    pkgPolicy Plcy(&Cache);
    if (ReadPinFile(Plcy) == false)
       return false;
-   
-   pkgCache::VerFile **VFList = new pkgCache::VerFile *[Cache.HeaderP->PackageCount];
-   memset(VFList,0,sizeof(*VFList)*Cache.HeaderP->PackageCount);
+
+   // Make sure we have a sentinel for the list.
+   unsigned long Count = Cache.HeaderP->PackageCount + 1;   
+   pkgCache::VerFile **VFList = new pkgCache::VerFile *[Count];
+   memset(VFList,0,sizeof(*VFList)*Count);
    
    // Map versions that we want to write out onto the VerList array.
    for (pkgCache::PkgIterator P = Cache.PkgBegin(); P.end() == false; P++)
@@ -426,7 +428,7 @@
       VFList[P->ID] = VF;
    }
    
-   LocalitySort(VFList,Cache.HeaderP->PackageCount,sizeof(*VFList));
+   LocalitySort(VFList,Count,sizeof(*VFList));
 
    // Iterate over all the package files and write them out.
    char *Buffer = new char[Cache.HeaderP->MaxVerFileSize+10];
diff -ruN apt/cmdline/apt-get.cc apt-patched/cmdline/apt-get.cc
--- apt/cmdline/apt-get.cc	Sun Mar  4 03:36:26 2001
+++ apt-patched/cmdline/apt-get.cc	Wed Aug 22 18:26:23 2001
@@ -103,6 +103,8 @@
       c1out << Yes << endl;
       return true;
    }
+
+   fflush(NULL);
    
    char C = 0;
    char Jnk = 0;
diff -ruN apt/configure apt-patched/configure
--- apt/configure	Thu Mar  8 03:22:14 2001
+++ apt-patched/configure	Wed Aug 22 18:26:23 2001
@@ -2374,7 +2374,7 @@
 
 ac_given_srcdir=$srcdir
 
-trap 'rm -fr `echo "environment.mak:buildlib/environment.mak.in makefile:buildlib/makefile.in include/config.h:buildlib/config.h.in include/apti18n.h:buildlib/apti18n.h.in" | sed "s/:[^ ]*//g"` conftest*; exit 1' 1 2 15
+trap 'rm -fr `echo "environment.mak:buildlib/environment.mak.in makefile.wrap:buildlib/makefile.in include/config.h:buildlib/config.h.in include/apti18n.h:buildlib/apti18n.h.in" | sed "s/:[^ ]*//g"` conftest*; exit 1' 1 2 15
 EOF
 cat >> $CONFIG_STATUS <<EOF
 
@@ -2483,7 +2483,7 @@
 
 cat >> $CONFIG_STATUS <<EOF
 
-CONFIG_FILES=\${CONFIG_FILES-"environment.mak:buildlib/environment.mak.in makefile:buildlib/makefile.in"}
+CONFIG_FILES=\${CONFIG_FILES-"environment.mak:buildlib/environment.mak.in makefile.wrap:buildlib/makefile.in"}
 EOF
 cat >> $CONFIG_STATUS <<\EOF
 for ac_file in .. $CONFIG_FILES; do if test "x$ac_file" != x..; then
@@ -2649,7 +2649,7 @@
 
 EOF
 cat >> $CONFIG_STATUS <<\EOF
-make -s dirs
+make -f makefile.wrap -s dirs
 exit 0
 EOF
 chmod +x $CONFIG_STATUS
diff -ruN apt/configure.in apt-patched/configure.in
--- apt/configure.in	Thu Mar  8 03:22:13 2001
+++ apt-patched/configure.in	Wed Aug 22 18:26:23 2001
@@ -162,4 +162,4 @@
 ah_GLIBC_VER
 ah_LIBSTDCPP_VER
 
-AC_OUTPUT(environment.mak:buildlib/environment.mak.in makefile:buildlib/makefile.in,make -s dirs)
+AC_OUTPUT(environment.mak:buildlib/environment.mak.in makefile.wrap:buildlib/makefile.in,make -f makefile.wrap -s dirs)
diff -ruN apt/dselect/install apt-patched/dselect/install
--- apt/dselect/install	Tue Feb 20 08:03:17 2001
+++ apt-patched/dselect/install	Wed Aug 22 18:26:23 2001
@@ -3,8 +3,8 @@
 # Get the configuration from /etc/apt/apt.conf
 CLEAN="prompt"
 OPTS="-f"
-APTGET="/usr/bin/apt-get"
-DPKG="/usr/bin/dpkg"
+APTGET="@PREFIX@/bin/apt-get"
+DPKG="@PREFIX@/bin/dpkg"
 DPKG_OPTS="--admindir=$1"
 APT_OPT0="-oDir::State::status=$1/status"
 APT_OPT1="-oDPkg::Options::=$DPKG_OPTS"
diff -ruN apt/dselect/setup apt-patched/dselect/setup
--- apt/dselect/setup	Thu Jan 27 05:15:10 2000
+++ apt-patched/dselect/setup	Wed Aug 22 18:26:28 2001
@@ -23,15 +23,17 @@
 my $vardir=$ARGV[0];
 my $method=$ARGV[1];
 my $option=$ARGV[2];
-my $config_file = '/etc/apt/sources.list';
+my $config_file = '@PREFIX@/etc/apt/sources.list';
 
-my $boldon=`setterm -bold on`;
-my $boldoff=`setterm -bold off`;
+my $boldon=`setterm -bold on 2>/dev/null`;
+my $boldoff=`setterm -bold off 2>/dev/null`;
+$boldon = "" unless defined $boldon;
+$boldoff = "" unless defined $boldon;
 
 my @known_types           = ('deb');
 my @known_access         = ('http', 'ftp', 'file');
-my @typical_distributions = ('stable', 'unstable', 'frozen', 'non-US');
-my @typical_components    = ('main', 'contrib', 'non-free');
+my @typical_distributions = ('stable', 'unstable');
+my @typical_components    = ('main', 'crypto');
 
 my %known_access           = map {($_,$_)} @known_access;
 my %typical_distributions  = map {($_,$_)} @typical_distributions;
@@ -120,7 +122,7 @@
   $type         = 'deb';
   $urn          = "http://http.us.debian.org/debian" unless $urn;
   $distribution = "stable" unless $distribution;
-  $components   = "main contrib non-free" unless $components;
+  $components   = "main crypto" unless $components;
 
     
   $rec->{'Type'} = 'deb';
diff -ruN apt/dselect/update apt-patched/dselect/update
--- apt/dselect/update	Tue Feb 20 08:03:18 2001
+++ apt-patched/dselect/update	Wed Aug 22 18:26:23 2001
@@ -3,13 +3,13 @@
 
 # Get the configuration from /etc/apt/apt.conf
 OPTS="-f"
-APTGET="/usr/bin/apt-get"
-APTCACHE="/usr/bin/apt-cache"
-DPKG="/usr/bin/dpkg"
+APTGET="@PREFIX@/bin/apt-get"
+APTCACHE="@PREFIX@/bin/apt-cache"
+DPKG="@PREFIX@/bin/dpkg"
 DPKG_OPTS="--admindir=$1"
 APT_OPT0="-oDir::State::status=$1/status"
 APT_OPT1="-oDPkg::Options::=$DPKG_OPTS"
-CACHEDIR="/var/cache/apt"
+CACHEDIR="@PREFIX@/var/cache/apt"
 PROMPT="false"
 RES=`apt-config shell OPTS DSelect::UpdateOptions \
       DPKG Dir::Bin::dpkg/f APTGET Dir::Bin::apt-get/f \
diff -ruN apt/methods/connect.cc apt-patched/methods/connect.cc
--- apt/methods/connect.cc	Tue Feb 20 08:03:18 2001
+++ apt-patched/methods/connect.cc	Wed Aug 22 18:26:23 2001
@@ -90,7 +90,7 @@
 
    // Check the socket for an error condition
    unsigned int Err;
-   unsigned int Len = sizeof(Err);
+   int Len = sizeof(Err);
    if (getsockopt(Fd,SOL_SOCKET,SO_ERROR,&Err,&Len) != 0)
       return _error->Errno("getsockopt","Failed");
    
diff -ruN apt/methods/ftp.cc apt-patched/methods/ftp.cc
--- apt/methods/ftp.cc	Thu Mar  8 03:20:43 2001
+++ apt-patched/methods/ftp.cc	Wed Aug 22 18:26:23 2001
@@ -688,7 +688,7 @@
       if (WaitFd(DataFd,true,TimeOut) == false)
 	 return _error->Error("Could not connect data socket, connection timed out");
       unsigned int Err;
-      unsigned int Len = sizeof(Err);
+      int Len = sizeof(Err);
       if (getsockopt(DataFd,SOL_SOCKET,SO_ERROR,&Err,&Len) != 0)
 	 return _error->Errno("getsockopt","Failed");
       if (Err != 0)
@@ -733,7 +733,7 @@
    
    // Determine the name to send to the remote
    struct sockaddr_storage Addr;
-   socklen_t AddrLen = sizeof(Addr);
+   int AddrLen = sizeof(Addr);
    if (getsockname(DataListenFd,(sockaddr *)&Addr,&AddrLen) < 0)
       return _error->Errno("getsockname","Could not determine the socket's name");
 
@@ -809,7 +809,7 @@
       
    // Accept the connection
    struct sockaddr_in Addr;
-   socklen_t Len = sizeof(Addr);
+   int Len = sizeof(Addr);
    DataFd = accept(DataListenFd,(struct sockaddr *)&Addr,&Len);
    if (DataFd < 0)
       return _error->Errno("accept","Unable to accept connection");
diff -ruN apt/methods/ftp.h apt-patched/methods/ftp.h
--- apt/methods/ftp.h	Thu Mar  8 03:20:43 2001
+++ apt-patched/methods/ftp.h	Wed Aug 22 18:26:23 2001
@@ -26,11 +26,11 @@
    
    // Generic Peer Address
    struct sockaddr_storage PeerAddr;
-   socklen_t PeerAddrLen;
+   int PeerAddrLen;
    
    // Generic Server Address (us)
    struct sockaddr_storage ServerAddr;
-   socklen_t ServerAddrLen;
+   int ServerAddrLen;
    
    // Private helper functions
    bool ReadLine(string &Text);      
diff -ruN apt/methods/rfc2553emu.h apt-patched/methods/rfc2553emu.h
--- apt/methods/rfc2553emu.h	Sun Jun 18 08:04:45 2000
+++ apt-patched/methods/rfc2553emu.h	Wed Aug 22 18:26:23 2001
@@ -26,6 +26,11 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 
+// Always use full emulation on Darwin:
+//  netdb.h has the structures and constants, but getnameinfo() is missing
+//  and getaddrinfo() seems to be broken
+#ifndef __APPLE__
+
 // Autosense getaddrinfo
 #if defined(AI_PASSIVE) && defined(EAI_NONAME)
 #define HAVE_GETADDRINFO
@@ -36,6 +41,8 @@
 #define HAVE_GETNAMEINFO
 #endif
 
+#endif /* __APPLE__ */
+
 // getaddrinfo support?
 #ifndef HAVE_GETADDRINFO
   // Renamed to advoid type clashing.. (for debugging)
@@ -100,6 +107,9 @@
 //  #define NI_NOFQDN (1<<2)
   #define NI_NAMEREQD (1<<3)
   #define NI_DATAGRAM (1<<4)
+  #endif
+  #ifndef NI_DATAGRAM
+  #define NI_DATAGRAM NI_DGRAM
   #endif
 
   #define sockaddr_storage sockaddr_in
diff -ruN apt/patch_flush apt-patched/patch_flush
--- apt/patch_flush	Thu Jan  1 01:00:00 1970
+++ apt-patched/patch_flush	Wed Aug 22 18:49:33 2001
@@ -0,0 +1,11 @@
+#!/bin/sh
+set -e
+
+files=`find . -name '*.cc' -print | xargs grep -l 'flush;'`
+
+for i in $files ; do
+  sed 's/<< flush;/<< flush, fflush(NULL);/g' <$i >$i.tmp
+  mv $i.tmp $i
+done
+
+exit 0
diff -ruN apt/sources.list.fink apt-patched/sources.list.fink
--- apt/sources.list.fink	Thu Jan  1 01:00:00 1970
+++ apt-patched/sources.list.fink	Wed Aug 22 18:55:44 2001
@@ -0,0 +1,16 @@
+# Default APT sources configuration for Fink
+
+# Official binary distribution: download location for packages
+# from the latest release
+deb http://fink.sourceforge.net/bindist release main crypto
+
+# Official binary distribution: download location for updated
+# packages built between releases
+deb http://fink.sourceforge.net/bindist current main crypto
+
+# Local package trees - packages built from source locally
+# NOTE: keep this in sync with the Trees: line in @PREFIX@/etc/fink.conf
+# NOTE: run 'fink scanpackages' to create Packages.gz files
+deb file:@PREFIX@/fink local main
+deb file:@PREFIX@/fink stable main crypto bootstrap
+#deb file:@PREFIX@/fink unstable main crypto
