diff -ruN gettext-0.10.40/configure gettext-0.10.40-patched/configure
--- gettext-0.10.40/configure	Sat Sep 15 02:47:29 2001
+++ gettext-0.10.40-patched/configure	Sat Sep 22 20:50:49 2001
@@ -3782,7 +3782,7 @@
     ;;
 
   darwin* | rhapsody*)
-    allow_undefined_flag='-undefined suppress'
+    allow_undefined_flag='-flat_namespace -undefined suppress'
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
