diff -ru ncurses-5.2/c++/cursesw.cc ncurses-5.2-patched/c++/cursesw.cc
--- ncurses-5.2/c++/cursesw.cc	Sat Sep  2 20:55:31 2000
+++ ncurses-5.2-patched/c++/cursesw.cc	Thu Jun  6 20:31:51 2002
@@ -51,8 +51,9 @@
     char buf[BUFSIZ];
     int result = wgetstr(w, buf);
     if (result == OK) {
-	strstreambuf ss(buf, sizeof(buf));
-	result = ss.vscan(fmt, (_IO_va_list)args);
+//	strstreambuf ss(buf, sizeof(buf));
+//	result = ss.vscan(fmt, (_IO_va_list)args);
+	vsscanf(buf, fmt, args);
     }
     va_end(args);
     return result;
@@ -73,8 +74,9 @@
     if (result == OK) {
 	result = wgetstr(w, buf);
 	if (result == OK) {
-	    strstreambuf ss(buf, sizeof(buf));
-	    result = ss.vscan(fmt, (_IO_va_list)args);
+//	    strstreambuf ss(buf, sizeof(buf));
+//	    result = ss.vscan(fmt, (_IO_va_list)args);
+	    vsscanf(buf, fmt, args);
 	}
     }
     va_end(args);
diff -ru ncurses-5.2/configure ncurses-5.2-patched/configure
--- ncurses-5.2/configure	Sat Oct 21 00:58:49 2000
+++ ncurses-5.2-patched/configure	Thu Jun  6 16:44:29 2002
@@ -2688,6 +2688,11 @@
 			MK_SHARED_LIB='$(LD) -Bshareable -o $@'
 		fi
 		;;
+	darwin*)
+		CC_SHARED_OPTS='-fno-common'
+		test "$cf_cv_shlib_version" = auto && cf_cv_shlib_version=rel
+		MK_SHARED_LIB='$(CC) -dynamiclib -install_name $(libdir)/`basename $@ .$(REL_VERSION)`.$(ABI_VERSION) -compatibility_version $(ABI_VERSION) -current_version $(REL_VERSION) -o $@'
+		;;
 	osf*|mls+*)
 		# tested with OSF/1 V3.2 and 'cc'
 		# tested with OSF/1 V3.2 and gcc 2.6.3 (but the c++ demo didn't
@@ -7057,8 +7062,10 @@
 	shared)
 		case $cf_cv_system_name in
 		hpux*)	DFT_DEP_SUFFIX='.sl'  ;;
+		darwin*) DFT_DEP_SUFFIX='.dylib'  ;;
 		*)	DFT_DEP_SUFFIX='.so'  ;;
 		esac
+		DFT_DEP_SUFFIX="${DFT_DEP_SUFFIX}."'$(REL_VERSION)'
 	esac
 	test -n "$LIB_SUFFIX" && DFT_DEP_SUFFIX="${LIB_SUFFIX}${DFT_DEP_SUFFIX}"
 echo "$ac_t""$DFT_DEP_SUFFIX" 1>&6
@@ -7091,6 +7098,7 @@
 	shared)
 		case $cf_cv_system_name in
 		hpux*)	CXX_LIB_SUFFIX='.sl'  ;;
+		darwin*) CXX_LIB_SUFFIX='.dylib'  ;;
 		*)	CXX_LIB_SUFFIX='.so'  ;;
 		esac
 	esac
@@ -7767,6 +7775,7 @@
 	shared)
 		case $cf_cv_system_name in
 		hpux*)	cf_suffix='.sl'  ;;
+		darwin*) cf_suffix='.dylib'  ;;
 		*)	cf_suffix='.so'  ;;
 		esac
 	esac
@@ -7818,6 +7827,7 @@
 	shared)
 		case $cf_cv_system_name in
 		hpux*)	cf_suffix='.sl'  ;;
+		darwin*) cf_suffix='.dylib'  ;;
 		*)	cf_suffix='.so'  ;;
 		esac
 	esac
diff -ru ncurses-5.2/misc/terminfo.src ncurses-5.2-patched/misc/terminfo.src
--- ncurses-5.2/misc/terminfo.src	Sun Sep 24 04:29:16 2000
+++ ncurses-5.2-patched/misc/terminfo.src	Thu Jun  6 16:44:29 2002
@@ -12955,7 +12955,7 @@
 	kend=\E[1~, khome=\E[H, kich1=\E[2~, knp=\E[6~, kpp=\E[5~, 
 	lf1=pf1, lf2=pf2, lf3=pf3, lf4=pf4, rmkx=\E>, smkx=\E=, 
 	use=ncr260vt300an, 
-NCR260VT300WPP|ncr260vt300wpp|NCR 2900_260 vt300 wide mode pc+  kybd, 
+ncr260vt300wpp|NCR 2900_260 vt300 wide mode pc+  kybd, 
 	cols#132, 
 	cup=\E[%i%p1%d;%p2%dH$<30>, 
 	is2=\E[!p\E[?3;7;19;67h\E[?1;4l\E[1;0%w\E(B\E)0\017\E[2J\E[1;1H\E>$<200>, 
diff -ru ncurses-5.2/ncurses/tinfo/write_entry.c ncurses-5.2-patched/ncurses/tinfo/write_entry.c
--- ncurses-5.2/ncurses/tinfo/write_entry.c	Wed Oct  4 04:32:14 2000
+++ ncurses-5.2-patched/ncurses/tinfo/write_entry.c	Thu Jun  6 16:44:29 2002
@@ -319,6 +319,9 @@
 #endif
 	    if (code != 0 && errno == ENOENT)
 		code = 0;
+	    if (code != 0) {
+	      printf("error from remove/unlink: code %d, errno %d\n", code, errno);
+	    }
 #if USE_SYMLINKS
 	    if (symlink(symlinkname, linkname) < 0)
 #else
@@ -331,6 +334,7 @@
 		 * target, then the source must be on a filesystem
 		 * that uses caseless filenames, such as Win32, etc.
 		 */
+	        printf("error from link/symlink: code %d, errno %d\n", code, errno);
 		if (code == 0 && errno == EEXIST)
 		    _nc_warning("can't link %s to %s", filename, linkname);
 		else if (code == 0 && errno == EPERM)
